{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: unidecode in c:\\conda\\lib\\site-packages (1.2.0)\n",
      "Requirement already satisfied: gensim in c:\\conda\\lib\\site-packages (3.8.3)\n",
      "Requirement already satisfied: smart-open>=1.8.1 in c:\\conda\\lib\\site-packages (from gensim) (3.0.0)\n",
      "Requirement already satisfied: six>=1.5.0 in c:\\conda\\lib\\site-packages (from gensim) (1.14.0)\n",
      "Requirement already satisfied: scipy>=0.18.1 in c:\\conda\\lib\\site-packages (from gensim) (1.4.1)\n",
      "Requirement already satisfied: numpy>=1.11.3 in c:\\conda\\lib\\site-packages (from gensim) (1.18.4)\n",
      "Requirement already satisfied: Cython==0.29.14 in c:\\conda\\lib\\site-packages (from gensim) (0.29.14)\n",
      "Requirement already satisfied: requests in c:\\conda\\lib\\site-packages (from smart-open>=1.8.1->gensim) (2.22.0)\n",
      "Requirement already satisfied: idna<2.9,>=2.5 in c:\\conda\\lib\\site-packages (from requests->smart-open>=1.8.1->gensim) (2.8)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\conda\\lib\\site-packages (from requests->smart-open>=1.8.1->gensim) (2020.12.5)\n",
      "Requirement already satisfied: chardet<3.1.0,>=3.0.2 in c:\\conda\\lib\\site-packages (from requests->smart-open>=1.8.1->gensim) (3.0.4)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in c:\\conda\\lib\\site-packages (from requests->smart-open>=1.8.1->gensim) (1.25.8)\n",
      "Requirement already satisfied: flashtext in c:\\conda\\lib\\site-packages (2.7)\n"
     ]
    }
   ],
   "source": [
    "!pip install unidecode\n",
    "!pip install gensim\n",
    "!pip install flashtext"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import re\n",
    "import traceback \n",
    "from pyspark.sql import SparkSession\n",
    "from elasticsearch import Elasticsearch, helpers\n",
    "import requests, json, os, csv\n",
    "from nltk.corpus import stopwords\n",
    "import string\n",
    "import unidecode\n",
    "from nltk.tokenize import sent_tokenize\n",
    "from nltk.tokenize import word_tokenize\n",
    "import nltk\n",
    "import pandas as pd\n",
    "from flashtext import KeywordProcessor\n",
    "import spacy\n",
    "import pytextrank\n",
    "from sklearn.feature_extraction.text import TfidfTransformer,TfidfVectorizer\n",
    "from nltk.stem import WordNetLemmatizer\n",
    "import pprint\n",
    "from IPython.display import display "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "s_top_words = set(stopwords.words('english'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data loading"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_data(path = 'metadata.csv'):\n",
    "    spark = SparkSession \\\n",
    "    .builder \\\n",
    "    .appName(\"ElasticSpark-1\") \\\n",
    "    .config(\"spark.driver.extraClassPath\", \"/path/elasticsearch-hadoop-7.6.2/dist/elasticsearch-spark-20_2.11-7.6.2.jar\") \\\n",
    "    .config(\"spark.es.port\",\"9200\") \\\n",
    "    .config(\"spark.driver.memory\", \"8G\") \\\n",
    "    .config(\"spark.executor.memory\", \"12G\") \\\n",
    "    .getOrCreate()\n",
    "    metadata_df = spark.read.csv(path, multiLine=True, header=True)\n",
    "    metadata_df.show(1)\n",
    "    metadata_df = metadata_df.select(\"*\").limit(1000)\n",
    "    metadata_df.show(3)\n",
    "    metadata_table = metadata_df.toPandas()\n",
    "    print(\"Data loaded.\")\n",
    "    display(metadata_table)\n",
    "    metadata_table[\"pr_title\"]=metadata_table[\"title\"]\n",
    "    metadata_table[\"pr_abstract\"]=metadata_table[\"abstract\"]\n",
    "    return metadata_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_index(es_index=\"covid\"):\n",
    "    res = requests.get('http://localhost:9200')\n",
    "    print (res.content)\n",
    "    es = Elasticsearch([{'host': 'localhost', 'port': '9200'}])\n",
    "    mapping = {\n",
    "        'settings':{\n",
    "            'number_of_shards': 1, \n",
    "            'number_of_replicas': 1\n",
    "        },\n",
    "        'mappings': {\n",
    "            'properties': {\n",
    "                'cord_uid': {\n",
    "                    'index': 'true', \n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'sha': {\n",
    "                    'index': 'true', \n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'source_x': {\n",
    "                    'index': 'true', \n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'title': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text', \n",
    "                    'similarity': 'BM25'\n",
    "                },\n",
    "                'pr_title': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text', \n",
    "                    'similarity': 'BM25'\n",
    "                },\n",
    "                'doi': {\n",
    "                    'index': 'true', \n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'pmcid': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'license': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'abstract': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text',\n",
    "                    'similarity': 'BM25'\n",
    "                },\n",
    "                'pr_abstract': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text',\n",
    "                    'similarity': 'BM25'\n",
    "                },\n",
    "                'publish_time': {\n",
    "                    'index': 'true', \n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'authors': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'journal': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'who_covidence_id': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'arxiv_id': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'pdf_json_files': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'pmc_json_files': {\n",
    "                    'index': 'true', \n",
    "                    'type': 'text'\n",
    "                },\n",
    "                'url': {\n",
    "                    'index': 'true',\n",
    "                    'type': 'text'\n",
    "                },\n",
    "                's2_id': {\n",
    "                    'index': 'true', \n",
    "                    'type': 'text'\n",
    "                }\n",
    "             }\n",
    "         }\n",
    "    }\n",
    "    if es.indices.exists(es_index):\n",
    "        es.indices.delete(es_index) \n",
    "\n",
    "    es.indices.create(index=es_index,body=mapping)\n",
    "    return es"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sentence Splitting, Tokenization and Normalization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "class TextNormalizer:\n",
    "    def __init__(self):\n",
    "        self.punctuation_table = str.maketrans('','',string.punctuation)\n",
    "\n",
    "    def normalize_text(self,text):\n",
    "        if text==None:\n",
    "            return None\n",
    "        try: \n",
    "            normalized_sentences = []\n",
    "            text = re.sub(' +',' ', text)\n",
    "            text = unidecode.unidecode(text)\n",
    "            text = text.lower()\n",
    "            sentences = sent_tokenize(text)\n",
    "        except:\n",
    "            print(\"ERROR:\", text)\n",
    "            traceback.print_exc()\n",
    "            return None\n",
    "        \n",
    "        for sentence in sentences:\n",
    "            #remove punctuation\n",
    "            sentence=re.sub(\"[\"+string.punctuation+\"\\d*]\",\" \",sentence)\n",
    "            #strip leading/trailing whitespace\n",
    "            sentence = sentence.strip()\n",
    "            words = word_tokenize(sentence)\n",
    "            new_sentence = ' '.join(words) #we want to keep it as before to extract phrases\n",
    "            normalized_sentences.append(new_sentence)\n",
    "        return normalized_sentences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def normalize_table(metadata_table):\n",
    "    normaliser = TextNormalizer()\n",
    "    \n",
    "    table_to_process=metadata_table[[\"pr_title\",\"pr_abstract\"]]\n",
    "    table_to_process[\"pr_title\"]=table_to_process[\"pr_title\"].apply(lambda x: normaliser.normalize_text(x))\n",
    "    table_to_process[\"pr_abstract\"]=table_to_process[\"pr_abstract\"].apply(lambda x: normaliser.normalize_text(x))\n",
    "    \n",
    "    for i in range(0, len(table_to_process)):\n",
    "        metadata_table.loc[i,\"pr_title\"] = table_to_process.loc[i,\"pr_title\"]\n",
    "        metadata_table.loc[i,\"pr_abstract\"] = table_to_process.loc[i,\"pr_abstract\"]\n",
    "    return metadata_table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Selecting key words"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def remove_stop_words(text):\n",
    "    if text==None:\n",
    "        return\n",
    "    for index,sentence in enumerate(text):\n",
    "        sentence = sentence.split(\" \") #performing tokenisation\n",
    "        sentence = [word for word in sentence if word not in s_top_words and len(word)>2]\n",
    "        sentence=\" \".join(sentence)\n",
    "        text[index]=sentence\n",
    "    return text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_words_corpus(table):\n",
    "    words_corpus=[]\n",
    "    for i in range(0, len(table)):\n",
    "        row=table.loc[i]\n",
    "        title_sentences = row[\"pr_title\"]\n",
    "        abstract_sentences = row[\"pr_abstract\"]\n",
    "        \n",
    "        if title_sentences!=None:\n",
    "            for i in range(0,len(title_sentences)):\n",
    "                words_corpus.extend(title_sentences[i].split())\n",
    "                \n",
    "        if  abstract_sentences!=None:\n",
    "            for i in range(0,len(abstract_sentences)):\n",
    "                words_corpus.extend(abstract_sentences[i].split())\n",
    "    return words_corpus\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_keywords_by_textrank(sentences):\n",
    "    if sentences==None:\n",
    "        return None\n",
    "    keywords=dict()\n",
    "    nlp = spacy.load('en_core_web_sm')\n",
    "    nlp.add_pipe(\"textrank\", last=True)\n",
    "    doc = nlp(\" \".join(sentences))\n",
    "\n",
    "    # examine the top-ranked phrases in the document\n",
    "\n",
    "    for p in doc._.phrases:\n",
    "        if p.rank>=0.05:\n",
    "            keywords[p.text]=p.rank\n",
    "    #         print(\"{:.4f} {:5d}  {}\".format(p.rank, p.count, p.text))\n",
    "    #         print(p.text)\n",
    "    return keywords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_keywords(text,keyword_processor):\n",
    "    sentences=[]\n",
    "    if text==None:\n",
    "        return None\n",
    "    for i in range(0, len(text)):\n",
    "        keywords_found = keyword_processor.extract_keywords(text[i])\n",
    "        sentences.append(\" \".join(keywords_found))\n",
    "    return sentences\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_two_keywords_methods(sentences, text_rank_key_word_processor, frequent_key_words_processor):\n",
    "    if sentences==None:\n",
    "        return None\n",
    "    text_rank_version = extract_keywords(sentences,text_rank_key_word_processor)\n",
    "    frequent_key_words_version = extract_keywords(sentences,frequent_key_words_processor)\n",
    "    intersect = set(frequent_key_words_version) - set(text_rank_version)\n",
    "\n",
    "    merged_version = text_rank_version + list(intersect)\n",
    "    return merged_version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def retain_best_tf_idf_keywords(sentences, index, tfIdf,tfIdfVectorizer):\n",
    "    if sentences==None:\n",
    "        return None\n",
    "    tf_idf_keyword_processor = KeywordProcessor()\n",
    "    df = pd.DataFrame(tfIdf[index].T.todense(), index=tfIdfVectorizer.get_feature_names(), columns=[\"TF_IDF\"])\n",
    "    df = df.sort_values('TF_IDF', ascending=False)\n",
    "    df = df[df.TF_IDF>0.09]\n",
    "    tf_idf_dict=df.T.to_dict('list')    \n",
    "    for keyword in tf_idf_dict.keys():\n",
    "        parts = \" \".join(keyword.split(\"_\"))\n",
    "        tf_idf_keyword_processor.add_keyword(keyword,parts)\n",
    "    sentences = extract_keywords(sentences,tf_idf_keyword_processor)\n",
    "    return sentences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def select_best_keywords(metadata_table):\n",
    "    table_to_process=metadata_table[[\"pr_title\",\"pr_abstract\"]]\n",
    "    table_to_process[\"pr_title\"]=table_to_process[\"pr_title\"].apply(lambda x: remove_stop_words(x))\n",
    "    table_to_process[\"pr_abstract\"]=table_to_process[\"pr_abstract\"].apply(lambda x: remove_stop_words(x))\n",
    "    \n",
    "    print(\"Text Data after removing of stop-words\")\n",
    "    display(table_to_process)\n",
    "\n",
    "    words_corpus=get_words_corpus(table_to_process)\n",
    "    print(len(words_corpus))\n",
    "\n",
    "    dist = nltk.FreqDist(words_corpus) #Creating a distribution of words' frequencies\n",
    "    grams=dist.most_common(1000) #Obtaining the most frequent words\n",
    "    bigrams = nltk.collocations.BigramAssocMeasures()\n",
    "    trigrams = nltk.collocations.TrigramAssocMeasures()\n",
    "\n",
    "    bigramFinder = nltk.collocations.BigramCollocationFinder.from_words(words_corpus)\n",
    "    trigramFinder = nltk.collocations.TrigramCollocationFinder.from_words(words_corpus)\n",
    "\n",
    "    print(\"Showing first\",2000,\"top-freqent words in the corpus\")\n",
    "    grams = pd.DataFrame(grams) \n",
    "    grams.index = range(1,len(grams)+1)\n",
    "    grams.columns = [\"Word\", \"Frequency\"]\n",
    "    display(grams)\n",
    "    \n",
    "    bi_filter=7\n",
    "    print(\"Showing bigrams in the corpus found by Pointwise Mutual Information method\")\n",
    "    print(\"Applying frequency filter: a bigramm occurs more than\",bi_filter,\"times\")\n",
    "    bigramFinder.apply_freq_filter(bi_filter)\n",
    "    bigramPMITable = pd.DataFrame(list(bigramFinder.score_ngrams(bigrams.pmi)), columns=['bigram','PMI']).sort_values(by='PMI', ascending=False)\n",
    "    bigramPMITable[\"bigram\"]=bigramPMITable[\"bigram\"].apply(lambda x: ' '.join(x))\n",
    "    display(bigramPMITable)\n",
    "\n",
    "    tri_filter=5\n",
    "    print(\"Showing trigrams in the corpus found by Pointwise Mutual Information method\")\n",
    "    print(\"Applying frequency filter: a trigramm occurs more than\",tri_filter,\"times\")\n",
    "    trigramFinder.apply_freq_filter(tri_filter)\n",
    "    trigramPMITable = pd.DataFrame(list(trigramFinder.score_ngrams(trigrams.pmi)), columns=['trigram','PMI']).sort_values(by='PMI', ascending=False)\n",
    "    trigramPMITable[\"trigram\"]=trigramPMITable[\"trigram\"].apply(lambda x: ' '.join(x))\n",
    "    display(trigramPMITable)\n",
    "\n",
    "    gram_dict=grams.set_index('Word').T.to_dict('list')\n",
    "    bigramPMIDict=bigramPMITable.set_index('bigram').T.to_dict('list')\n",
    "    trigramPMIDict=trigramPMITable.set_index('trigram').T.to_dict('list')\n",
    "\n",
    "    keyword_processor = KeywordProcessor()\n",
    "    textrank_keyword_processor = KeywordProcessor()\n",
    "\n",
    "    gram_dict.update(bigramPMIDict)\n",
    "    bigramPMIDict.update(trigramPMIDict)\n",
    "\n",
    "#     print(gram_dict)\n",
    "    print(\"Extracting keywords from texts using Pointwise Mutual Information method and TextRank\")\n",
    "    text_rank_key_words=dict()\n",
    "    for i in range(0, len(table_to_process)):\n",
    "        sentences=table_to_process.loc[i,\"pr_abstract\"]\n",
    "        if sentences!=None:\n",
    "            keywords=get_keywords_by_textrank(sentences)\n",
    "            if keywords!=None:\n",
    "                text_rank_key_words.update(keywords)\n",
    "                print(\"Text\",i,\"- Done\")\n",
    "\n",
    "    for keyword in gram_dict.keys():\n",
    "        parts=keyword.split()\n",
    "        parts=\"_\".join(parts)\n",
    "        keyword_processor.add_keyword(keyword,parts)\n",
    "\n",
    "    for keyword in text_rank_key_words.keys():\n",
    "        parts=keyword.split()\n",
    "        parts=\"_\".join(parts)\n",
    "        textrank_keyword_processor.add_keyword(keyword,parts)\n",
    "\n",
    "    \n",
    "    print(len(keyword_processor.get_all_keywords()))\n",
    "    print(len(textrank_keyword_processor.get_all_keywords()))\n",
    "    print(len(text_rank_key_words))\n",
    "\n",
    "    table_to_process[\"pr_abstract\"]=table_to_process[\"pr_abstract\"].apply(lambda x: merge_two_keywords_methods(x, textrank_keyword_processor, keyword_processor))     \n",
    "\n",
    "    for i in range(0, len(table_to_process)):\n",
    "        metadata_table.loc[i,\"pr_title\"] = table_to_process.loc[i,\"pr_title\"]\n",
    "        metadata_table.loc[i,\"pr_abstract\"] = table_to_process.loc[i,\"pr_abstract\"]\n",
    "\n",
    "    print(\"Comparison of Text Data after Keywords Extraction using Pointwise Mutual Information method and TextRank\")\n",
    "    display(metadata_table[[\"title\",\"pr_title\",\"abstract\",\"pr_abstract\"]])\n",
    "\n",
    "    print(\"Extracting keywords from texts using TF/IDF\")\n",
    "    dataset = []\n",
    "    for i in range(0, len(table_to_process[\"pr_abstract\"])):\n",
    "        sentences = table_to_process.loc[i,\"pr_abstract\"]\n",
    "        if sentences!=None:\n",
    "            sentences=\" \".join(sentences)\n",
    "            dataset.append(sentences)\n",
    "\n",
    "    tfIdfVectorizer=TfidfVectorizer(use_idf=True)\n",
    "    tfIdf = tfIdfVectorizer.fit_transform(dataset)\n",
    "\n",
    "    index=0\n",
    "    for i in range(0,len(metadata_table)):\n",
    "        if table_to_process.loc[i,\"pr_abstract\"]==None:\n",
    "            continue\n",
    "        metadata_table.loc[i,\"pr_abstract\"]=retain_best_tf_idf_keywords(table_to_process.loc[i,\"pr_abstract\"], index,tfIdf,tfIdfVectorizer)\n",
    "        index+=1\n",
    "    return metadata_table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Stemming or Morphological Analysis (Lemmatisation) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def lemmatise_text(sentences):\n",
    "    if sentences==None:\n",
    "        return None\n",
    "    lemmatizer = WordNetLemmatizer()\n",
    "    for i in range(0, len(sentences)):\n",
    "        try:\n",
    "            if sentences[i] == \"\":\n",
    "                continue\n",
    "            words=sentences[i].split()\n",
    "            lemmatised_words = [lemmatizer.lemmatize(word) for word in words]\n",
    "            lemmatised_words = ' '.join(lemmatised_words)\n",
    "            sentences[i]=lemmatised_words\n",
    "        except:\n",
    "            print(sentences)\n",
    "            print(sentences[i])\n",
    "            traceback.print_exc()\n",
    "            break\n",
    "    return sentences\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Indexing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def index_table(es,metadata_table,es_index=\"covid\"):\n",
    "    for i in range(0,len(metadata_table)):\n",
    "        metadata_table.iloc[i].to_json(es_index+'.json')\n",
    "        f = open(es_index+'.json')\n",
    "        docket_content = f.read()\n",
    "        row=json.loads(docket_content)\n",
    "        try:\n",
    "            es.index(index=es_index, id=i, body=row)\n",
    "        except:\n",
    "            traceback.print_exc() \n",
    "            print(\"Error:\", \"row #\"+str(i))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Searching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def search(es, es_index=\"covid\",\n",
    "    query={\n",
    "          \"query\": {\n",
    "            \"match_phrase\":{\"publish_time\":\"2000-08-15\"}\n",
    "          }\n",
    "        }):\n",
    "    res = es.search(index=es_index, body=query)\n",
    "    documents=[]\n",
    "    for i in range(0, len(res['hits']['hits'])):\n",
    "        doc=res['hits']['hits'][i]['_source']\n",
    "        documents.append(doc)\n",
    "    return documents"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Running of the program"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_program(path = 'metadata.csv',es_index=\"covid\", query={\n",
    "              \"query\": {\n",
    "                \"match_phrase\":{\"publish_time\":\"2000-08-15\"}\n",
    "              }\n",
    "            }):\n",
    "    #Data loading\n",
    "    print(\"Data loading\")\n",
    "    metadata_table= load_data(path)\n",
    "    print(\"pr_title and pr_abstract columns have been added\")\n",
    "    display(metadata_table)\n",
    "    \n",
    "    #Indexing\n",
    "    print(\"Creating index -\",es_index)\n",
    "    es=create_index(es_index)\n",
    "    print(\"Indexing\")\n",
    "    index_table(es,metadata_table,es_index=es_index)\n",
    "    print(\"Data indexed.\")\n",
    "\n",
    "    #Sentence splitting, text tokenisation and normalisation\n",
    "    print(\"Sentence splitting, text tokenisation and normalisation\")\n",
    "    metadata_table=normalize_table(metadata_table)\n",
    "    print(\"Comparison of Text Data after Sentence splitting, text tokenisation and normalisation step\")\n",
    "    display(metadata_table[[\"title\",\"pr_title\",\"abstract\",\"pr_abstract\"]])\n",
    "\n",
    "    #Selecting keywords\n",
    "    print(\"Selecting keywords\")\n",
    "    metadata_table=select_best_keywords(metadata_table)\n",
    "    print(\"Comparison of Text Data after Selecting keywords step\")\n",
    "    display(metadata_table[[\"title\",\"pr_title\",\"abstract\",\"pr_abstract\"]])\n",
    "    \n",
    "    #Text lemmatisation\n",
    "    print(\"Text lemmatisation\")\n",
    "    metadata_table[\"pr_abstract\"]=metadata_table[\"pr_abstract\"].apply(lambda x: lemmatise_text(x))\n",
    "    metadata_table[\"pr_title\"]=metadata_table[\"pr_title\"].apply(lambda x: lemmatise_text(x))\n",
    "    print(\"Comparison of Text Data after Applied Lemmatisation\")\n",
    "    display(metadata_table[[\"title\",\"pr_title\",\"abstract\",\"pr_abstract\"]])\n",
    "\n",
    "    #Indexing\n",
    "    print(\"Creating index -\",es_index)\n",
    "    es=create_index(es_index)\n",
    "    print(\"Indexing\")\n",
    "    index_table(es,metadata_table,es_index=es_index)\n",
    "    \n",
    "    #Searching in ElasticSearch\n",
    "    print(\"Searching in ElasticSearch\")\n",
    "    documents=search(es, es_index=es_index, query=query)\n",
    "    print(\"Retrieved documents:\")\n",
    "    pprint.pprint(documents)\n",
    "    \n",
    "    return documents,es"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data loading\n",
      "+--------+--------------------+--------+--------------------+--------------------+--------+---------+-------+--------------------+------------+--------------------+--------------+------+----------------+--------+--------------------+--------------------+--------------------+-----+\n",
      "|cord_uid|                 sha|source_x|               title|                 doi|   pmcid|pubmed_id|license|            abstract|publish_time|             authors|       journal|mag_id|who_covidence_id|arxiv_id|      pdf_json_files|      pmc_json_files|                 url|s2_id|\n",
      "+--------+--------------------+--------+--------------------+--------------------+--------+---------+-------+--------------------+------------+--------------------+--------------+------+----------------+--------+--------------------+--------------------+--------------------+-----+\n",
      "|ug7v899j|d1aafb70c066a2068...|     PMC|Clinical features...|10.1186/1471-2334...|PMC35282| 11472636|  no-cc|OBJECTIVE: This r...|  2001-07-04|Madani, Tariq A; ...|BMC Infect Dis|  null|            null|    null|document_parses/p...|document_parses/p...|https://www.ncbi....| null|\n",
      "+--------+--------------------+--------+--------------------+--------------------+--------+---------+-------+--------------------+------------+--------------------+--------------+------+----------------+--------+--------------------+--------------------+--------------------+-----+\n",
      "only showing top 1 row\n",
      "\n",
      "+--------+--------------------+--------+--------------------+--------------------+--------+---------+-------+--------------------+------------+--------------------+--------------+------+----------------+--------+--------------------+--------------------+--------------------+-----+\n",
      "|cord_uid|                 sha|source_x|               title|                 doi|   pmcid|pubmed_id|license|            abstract|publish_time|             authors|       journal|mag_id|who_covidence_id|arxiv_id|      pdf_json_files|      pmc_json_files|                 url|s2_id|\n",
      "+--------+--------------------+--------+--------------------+--------------------+--------+---------+-------+--------------------+------------+--------------------+--------------+------+----------------+--------+--------------------+--------------------+--------------------+-----+\n",
      "|ug7v899j|d1aafb70c066a2068...|     PMC|Clinical features...|10.1186/1471-2334...|PMC35282| 11472636|  no-cc|OBJECTIVE: This r...|  2001-07-04|Madani, Tariq A; ...|BMC Infect Dis|  null|            null|    null|document_parses/p...|document_parses/p...|https://www.ncbi....| null|\n",
      "|02tnwd4m|6b0567729c2143a66...|     PMC|Nitric oxide: a p...|        10.1186/rr14|PMC59543| 11667967|  no-cc|Inflammatory dise...|  2000-08-15|Vliet, Albert van...|    Respir Res|  null|            null|    null|document_parses/p...|document_parses/p...|https://www.ncbi....| null|\n",
      "|ejv2xln0|06ced00a5fc042159...|     PMC|Surfactant protei...|        10.1186/rr19|PMC59549| 11667972|  no-cc|Surfactant protei...|  2000-08-25|     Crouch, Erika C|    Respir Res|  null|            null|    null|document_parses/p...|document_parses/p...|https://www.ncbi....| null|\n",
      "+--------+--------------------+--------+--------------------+--------------------+--------+---------+-------+--------------------+------------+--------------------+--------------+------+----------------+--------+--------------------+--------------------+--------------------+-----+\n",
      "only showing top 3 rows\n",
      "\n",
      "Data loaded.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>mag_id</th>\n",
       "      <th>who_covidence_id</th>\n",
       "      <th>arxiv_id</th>\n",
       "      <th>pdf_json_files</th>\n",
       "      <th>pmc_json_files</th>\n",
       "      <th>url</th>\n",
       "      <th>s2_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ug7v899j</td>\n",
       "      <td>d1aafb70c066a2068b02786f8929fd9c900897fb</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Clinical features of culture-proven Mycoplasma...</td>\n",
       "      <td>10.1186/1471-2334-1-6</td>\n",
       "      <td>PMC35282</td>\n",
       "      <td>11472636</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>OBJECTIVE: This retrospective chart review des...</td>\n",
       "      <td>2001-07-04</td>\n",
       "      <td>Madani, Tariq A; Al-Ghamdi, Aisha A</td>\n",
       "      <td>BMC Infect Dis</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/d1aafb70c066a2068b027...</td>\n",
       "      <td>document_parses/pmc_json/PMC35282.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>02tnwd4m</td>\n",
       "      <td>6b0567729c2143a66d737eb0a2f63f2dce2e5a7d</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n",
       "      <td>10.1186/rr14</td>\n",
       "      <td>PMC59543</td>\n",
       "      <td>11667967</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Inflammatory diseases of the respiratory tract...</td>\n",
       "      <td>2000-08-15</td>\n",
       "      <td>Vliet, Albert van der; Eiserich, Jason P; Cros...</td>\n",
       "      <td>Respir Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/6b0567729c2143a66d737...</td>\n",
       "      <td>document_parses/pmc_json/PMC59543.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ejv2xln0</td>\n",
       "      <td>06ced00a5fc04215949aa72528f2eeaae1d58927</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Surfactant protein-D and pulmonary host defense</td>\n",
       "      <td>10.1186/rr19</td>\n",
       "      <td>PMC59549</td>\n",
       "      <td>11667972</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Surfactant protein-D (SP-D) participates in th...</td>\n",
       "      <td>2000-08-25</td>\n",
       "      <td>Crouch, Erika C</td>\n",
       "      <td>Respir Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/06ced00a5fc04215949aa...</td>\n",
       "      <td>document_parses/pmc_json/PMC59549.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2b73a28n</td>\n",
       "      <td>348055649b6b8cf2b9a376498df9bf41f7123605</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Role of endothelin-1 in lung disease</td>\n",
       "      <td>10.1186/rr44</td>\n",
       "      <td>PMC59574</td>\n",
       "      <td>11686871</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n",
       "      <td>2001-02-22</td>\n",
       "      <td>Fagan, Karen A; McMurtry, Ivan F; Rodman, David M</td>\n",
       "      <td>Respir Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/348055649b6b8cf2b9a37...</td>\n",
       "      <td>document_parses/pmc_json/PMC59574.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9785vg6d</td>\n",
       "      <td>5f48792a5fa08bed9f56016f4981ae2ca6031b32</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Gene expression in epithelial cells in respons...</td>\n",
       "      <td>10.1186/rr61</td>\n",
       "      <td>PMC59580</td>\n",
       "      <td>11686888</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n",
       "      <td>2001-05-11</td>\n",
       "      <td>Domachowske, Joseph B; Bonville, Cynthia A; Ro...</td>\n",
       "      <td>Respir Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/5f48792a5fa08bed9f560...</td>\n",
       "      <td>document_parses/pmc_json/PMC59580.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>880nqc0f</td>\n",
       "      <td>ed2de3694f5580ea38f4adf24bd1c8b46862df1f</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Mannose-binding lectin deficiency and acute ex...</td>\n",
       "      <td>10.2147/copd.s33714</td>\n",
       "      <td>PMC3514010</td>\n",
       "      <td>23226013</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>BACKGROUND: Mannose-binding lectin is a collec...</td>\n",
       "      <td>2012-11-23</td>\n",
       "      <td>Albert, Richard K; Connett, John; Curtis, Jeff...</td>\n",
       "      <td>Int J Chron Obstruct Pulmon Dis</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/ed2de3694f5580ea38f4a...</td>\n",
       "      <td>document_parses/pmc_json/PMC3514010.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>01b0vnnm</td>\n",
       "      <td>2a7c951e191425fd9fa5ac108f07a1f02eb75872</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The changing phenotype of microglia from homeo...</td>\n",
       "      <td>10.1186/2047-9158-1-9</td>\n",
       "      <td>PMC3514090</td>\n",
       "      <td>23210447</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>It has been nearly a century since the early d...</td>\n",
       "      <td>2012-04-24</td>\n",
       "      <td>Luo, Xiao-Guang; Chen, Sheng-Di</td>\n",
       "      <td>Transl Neurodegener</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/2a7c951e191425fd9fa5a...</td>\n",
       "      <td>document_parses/pmc_json/PMC3514090.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>5b29wtim</td>\n",
       "      <td>854e623d1f875e4605b2ffd3f72599d063a56cc0</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Diversity of Salmonella spp. serovars isolated...</td>\n",
       "      <td>10.1186/1746-6148-8-201</td>\n",
       "      <td>PMC3514206</td>\n",
       "      <td>23098237</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>BACKGROUND: Salmonellosis in water buffalo (Bu...</td>\n",
       "      <td>2012-10-25</td>\n",
       "      <td>Borriello, Giorgia; Lucibelli, Maria G; Pescia...</td>\n",
       "      <td>BMC Vet Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/854e623d1f875e4605b2f...</td>\n",
       "      <td>document_parses/pmc_json/PMC3514206.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>z65xxk1f</td>\n",
       "      <td>3695704554777889f8232a9ea086df70bf17ff58</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Severe Childhood Malaria Syndromes Defined by ...</td>\n",
       "      <td>10.1371/journal.pone.0049778</td>\n",
       "      <td>PMC3514223</td>\n",
       "      <td>23226502</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>BACKGROUND: Cerebral malaria (CM) and severe m...</td>\n",
       "      <td>2012-12-04</td>\n",
       "      <td>Burté, Florence; Brown, Biobele J.; Orimadegun...</td>\n",
       "      <td>PLoS One</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/3695704554777889f8232...</td>\n",
       "      <td>document_parses/pmc_json/PMC3514223.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>iuglkdcp</td>\n",
       "      <td>5a054a570f409d5c2006369f19addc98d024b6c1</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Predicting pseudoknotted structures across two...</td>\n",
       "      <td>10.1093/bioinformatics/bts575</td>\n",
       "      <td>PMC3516145</td>\n",
       "      <td>23044552</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Motivation: Laboratory RNA structure determina...</td>\n",
       "      <td>2012-12-01</td>\n",
       "      <td>Sperschneider, Jana; Datta, Amitava; Wise, Mic...</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/5a054a570f409d5c20063...</td>\n",
       "      <td>document_parses/pmc_json/PMC3516145.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     cord_uid                                       sha source_x  \\\n",
       "0    ug7v899j  d1aafb70c066a2068b02786f8929fd9c900897fb      PMC   \n",
       "1    02tnwd4m  6b0567729c2143a66d737eb0a2f63f2dce2e5a7d      PMC   \n",
       "2    ejv2xln0  06ced00a5fc04215949aa72528f2eeaae1d58927      PMC   \n",
       "3    2b73a28n  348055649b6b8cf2b9a376498df9bf41f7123605      PMC   \n",
       "4    9785vg6d  5f48792a5fa08bed9f56016f4981ae2ca6031b32      PMC   \n",
       "..        ...                                       ...      ...   \n",
       "995  880nqc0f  ed2de3694f5580ea38f4adf24bd1c8b46862df1f      PMC   \n",
       "996  01b0vnnm  2a7c951e191425fd9fa5ac108f07a1f02eb75872      PMC   \n",
       "997  5b29wtim  854e623d1f875e4605b2ffd3f72599d063a56cc0      PMC   \n",
       "998  z65xxk1f  3695704554777889f8232a9ea086df70bf17ff58      PMC   \n",
       "999  iuglkdcp  5a054a570f409d5c2006369f19addc98d024b6c1      PMC   \n",
       "\n",
       "                                                 title  \\\n",
       "0    Clinical features of culture-proven Mycoplasma...   \n",
       "1    Nitric oxide: a pro-inflammatory mediator in l...   \n",
       "2      Surfactant protein-D and pulmonary host defense   \n",
       "3                 Role of endothelin-1 in lung disease   \n",
       "4    Gene expression in epithelial cells in respons...   \n",
       "..                                                 ...   \n",
       "995  Mannose-binding lectin deficiency and acute ex...   \n",
       "996  The changing phenotype of microglia from homeo...   \n",
       "997  Diversity of Salmonella spp. serovars isolated...   \n",
       "998  Severe Childhood Malaria Syndromes Defined by ...   \n",
       "999  Predicting pseudoknotted structures across two...   \n",
       "\n",
       "                               doi       pmcid pubmed_id license  \\\n",
       "0            10.1186/1471-2334-1-6    PMC35282  11472636   no-cc   \n",
       "1                     10.1186/rr14    PMC59543  11667967   no-cc   \n",
       "2                     10.1186/rr19    PMC59549  11667972   no-cc   \n",
       "3                     10.1186/rr44    PMC59574  11686871   no-cc   \n",
       "4                     10.1186/rr61    PMC59580  11686888   no-cc   \n",
       "..                             ...         ...       ...     ...   \n",
       "995            10.2147/copd.s33714  PMC3514010  23226013   no-cc   \n",
       "996          10.1186/2047-9158-1-9  PMC3514090  23210447   cc-by   \n",
       "997        10.1186/1746-6148-8-201  PMC3514206  23098237   cc-by   \n",
       "998   10.1371/journal.pone.0049778  PMC3514223  23226502   cc-by   \n",
       "999  10.1093/bioinformatics/bts575  PMC3516145  23044552   cc-by   \n",
       "\n",
       "                                              abstract publish_time  \\\n",
       "0    OBJECTIVE: This retrospective chart review des...   2001-07-04   \n",
       "1    Inflammatory diseases of the respiratory tract...   2000-08-15   \n",
       "2    Surfactant protein-D (SP-D) participates in th...   2000-08-25   \n",
       "3    Endothelin-1 (ET-1) is a 21 amino acid peptide...   2001-02-22   \n",
       "4    Respiratory syncytial virus (RSV) and pneumoni...   2001-05-11   \n",
       "..                                                 ...          ...   \n",
       "995  BACKGROUND: Mannose-binding lectin is a collec...   2012-11-23   \n",
       "996  It has been nearly a century since the early d...   2012-04-24   \n",
       "997  BACKGROUND: Salmonellosis in water buffalo (Bu...   2012-10-25   \n",
       "998  BACKGROUND: Cerebral malaria (CM) and severe m...   2012-12-04   \n",
       "999  Motivation: Laboratory RNA structure determina...   2012-12-01   \n",
       "\n",
       "                                               authors  \\\n",
       "0                  Madani, Tariq A; Al-Ghamdi, Aisha A   \n",
       "1    Vliet, Albert van der; Eiserich, Jason P; Cros...   \n",
       "2                                      Crouch, Erika C   \n",
       "3    Fagan, Karen A; McMurtry, Ivan F; Rodman, David M   \n",
       "4    Domachowske, Joseph B; Bonville, Cynthia A; Ro...   \n",
       "..                                                 ...   \n",
       "995  Albert, Richard K; Connett, John; Curtis, Jeff...   \n",
       "996                    Luo, Xiao-Guang; Chen, Sheng-Di   \n",
       "997  Borriello, Giorgia; Lucibelli, Maria G; Pescia...   \n",
       "998  Burté, Florence; Brown, Biobele J.; Orimadegun...   \n",
       "999  Sperschneider, Jana; Datta, Amitava; Wise, Mic...   \n",
       "\n",
       "                             journal mag_id who_covidence_id arxiv_id  \\\n",
       "0                     BMC Infect Dis   None             None     None   \n",
       "1                         Respir Res   None             None     None   \n",
       "2                         Respir Res   None             None     None   \n",
       "3                         Respir Res   None             None     None   \n",
       "4                         Respir Res   None             None     None   \n",
       "..                               ...    ...              ...      ...   \n",
       "995  Int J Chron Obstruct Pulmon Dis   None             None     None   \n",
       "996              Transl Neurodegener   None             None     None   \n",
       "997                      BMC Vet Res   None             None     None   \n",
       "998                         PLoS One   None             None     None   \n",
       "999                   Bioinformatics   None             None     None   \n",
       "\n",
       "                                        pdf_json_files  \\\n",
       "0    document_parses/pdf_json/d1aafb70c066a2068b027...   \n",
       "1    document_parses/pdf_json/6b0567729c2143a66d737...   \n",
       "2    document_parses/pdf_json/06ced00a5fc04215949aa...   \n",
       "3    document_parses/pdf_json/348055649b6b8cf2b9a37...   \n",
       "4    document_parses/pdf_json/5f48792a5fa08bed9f560...   \n",
       "..                                                 ...   \n",
       "995  document_parses/pdf_json/ed2de3694f5580ea38f4a...   \n",
       "996  document_parses/pdf_json/2a7c951e191425fd9fa5a...   \n",
       "997  document_parses/pdf_json/854e623d1f875e4605b2f...   \n",
       "998  document_parses/pdf_json/3695704554777889f8232...   \n",
       "999  document_parses/pdf_json/5a054a570f409d5c20063...   \n",
       "\n",
       "                                   pmc_json_files  \\\n",
       "0      document_parses/pmc_json/PMC35282.xml.json   \n",
       "1      document_parses/pmc_json/PMC59543.xml.json   \n",
       "2      document_parses/pmc_json/PMC59549.xml.json   \n",
       "3      document_parses/pmc_json/PMC59574.xml.json   \n",
       "4      document_parses/pmc_json/PMC59580.xml.json   \n",
       "..                                            ...   \n",
       "995  document_parses/pmc_json/PMC3514010.xml.json   \n",
       "996  document_parses/pmc_json/PMC3514090.xml.json   \n",
       "997  document_parses/pmc_json/PMC3514206.xml.json   \n",
       "998  document_parses/pmc_json/PMC3514223.xml.json   \n",
       "999  document_parses/pmc_json/PMC3516145.xml.json   \n",
       "\n",
       "                                                   url s2_id  \n",
       "0    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None  \n",
       "1    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  None  \n",
       "2    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  None  \n",
       "3    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  None  \n",
       "4    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  None  \n",
       "..                                                 ...   ...  \n",
       "995  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None  \n",
       "996  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None  \n",
       "997  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None  \n",
       "998  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None  \n",
       "999  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None  \n",
       "\n",
       "[1000 rows x 19 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pr_title and pr_abstract columns have been added\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>...</th>\n",
       "      <th>journal</th>\n",
       "      <th>mag_id</th>\n",
       "      <th>who_covidence_id</th>\n",
       "      <th>arxiv_id</th>\n",
       "      <th>pdf_json_files</th>\n",
       "      <th>pmc_json_files</th>\n",
       "      <th>url</th>\n",
       "      <th>s2_id</th>\n",
       "      <th>pr_title</th>\n",
       "      <th>pr_abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ug7v899j</td>\n",
       "      <td>d1aafb70c066a2068b02786f8929fd9c900897fb</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Clinical features of culture-proven Mycoplasma...</td>\n",
       "      <td>10.1186/1471-2334-1-6</td>\n",
       "      <td>PMC35282</td>\n",
       "      <td>11472636</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>OBJECTIVE: This retrospective chart review des...</td>\n",
       "      <td>2001-07-04</td>\n",
       "      <td>...</td>\n",
       "      <td>BMC Infect Dis</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/d1aafb70c066a2068b027...</td>\n",
       "      <td>document_parses/pmc_json/PMC35282.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "      <td>Clinical features of culture-proven Mycoplasma...</td>\n",
       "      <td>OBJECTIVE: This retrospective chart review des...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>02tnwd4m</td>\n",
       "      <td>6b0567729c2143a66d737eb0a2f63f2dce2e5a7d</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n",
       "      <td>10.1186/rr14</td>\n",
       "      <td>PMC59543</td>\n",
       "      <td>11667967</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Inflammatory diseases of the respiratory tract...</td>\n",
       "      <td>2000-08-15</td>\n",
       "      <td>...</td>\n",
       "      <td>Respir Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/6b0567729c2143a66d737...</td>\n",
       "      <td>document_parses/pmc_json/PMC59543.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>None</td>\n",
       "      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n",
       "      <td>Inflammatory diseases of the respiratory tract...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ejv2xln0</td>\n",
       "      <td>06ced00a5fc04215949aa72528f2eeaae1d58927</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Surfactant protein-D and pulmonary host defense</td>\n",
       "      <td>10.1186/rr19</td>\n",
       "      <td>PMC59549</td>\n",
       "      <td>11667972</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Surfactant protein-D (SP-D) participates in th...</td>\n",
       "      <td>2000-08-25</td>\n",
       "      <td>...</td>\n",
       "      <td>Respir Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/06ced00a5fc04215949aa...</td>\n",
       "      <td>document_parses/pmc_json/PMC59549.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>None</td>\n",
       "      <td>Surfactant protein-D and pulmonary host defense</td>\n",
       "      <td>Surfactant protein-D (SP-D) participates in th...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2b73a28n</td>\n",
       "      <td>348055649b6b8cf2b9a376498df9bf41f7123605</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Role of endothelin-1 in lung disease</td>\n",
       "      <td>10.1186/rr44</td>\n",
       "      <td>PMC59574</td>\n",
       "      <td>11686871</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n",
       "      <td>2001-02-22</td>\n",
       "      <td>...</td>\n",
       "      <td>Respir Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/348055649b6b8cf2b9a37...</td>\n",
       "      <td>document_parses/pmc_json/PMC59574.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>None</td>\n",
       "      <td>Role of endothelin-1 in lung disease</td>\n",
       "      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9785vg6d</td>\n",
       "      <td>5f48792a5fa08bed9f56016f4981ae2ca6031b32</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Gene expression in epithelial cells in respons...</td>\n",
       "      <td>10.1186/rr61</td>\n",
       "      <td>PMC59580</td>\n",
       "      <td>11686888</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n",
       "      <td>2001-05-11</td>\n",
       "      <td>...</td>\n",
       "      <td>Respir Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/5f48792a5fa08bed9f560...</td>\n",
       "      <td>document_parses/pmc_json/PMC59580.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>None</td>\n",
       "      <td>Gene expression in epithelial cells in respons...</td>\n",
       "      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>880nqc0f</td>\n",
       "      <td>ed2de3694f5580ea38f4adf24bd1c8b46862df1f</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Mannose-binding lectin deficiency and acute ex...</td>\n",
       "      <td>10.2147/copd.s33714</td>\n",
       "      <td>PMC3514010</td>\n",
       "      <td>23226013</td>\n",
       "      <td>no-cc</td>\n",
       "      <td>BACKGROUND: Mannose-binding lectin is a collec...</td>\n",
       "      <td>2012-11-23</td>\n",
       "      <td>...</td>\n",
       "      <td>Int J Chron Obstruct Pulmon Dis</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/ed2de3694f5580ea38f4a...</td>\n",
       "      <td>document_parses/pmc_json/PMC3514010.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "      <td>Mannose-binding lectin deficiency and acute ex...</td>\n",
       "      <td>BACKGROUND: Mannose-binding lectin is a collec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>01b0vnnm</td>\n",
       "      <td>2a7c951e191425fd9fa5ac108f07a1f02eb75872</td>\n",
       "      <td>PMC</td>\n",
       "      <td>The changing phenotype of microglia from homeo...</td>\n",
       "      <td>10.1186/2047-9158-1-9</td>\n",
       "      <td>PMC3514090</td>\n",
       "      <td>23210447</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>It has been nearly a century since the early d...</td>\n",
       "      <td>2012-04-24</td>\n",
       "      <td>...</td>\n",
       "      <td>Transl Neurodegener</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/2a7c951e191425fd9fa5a...</td>\n",
       "      <td>document_parses/pmc_json/PMC3514090.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "      <td>The changing phenotype of microglia from homeo...</td>\n",
       "      <td>It has been nearly a century since the early d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>5b29wtim</td>\n",
       "      <td>854e623d1f875e4605b2ffd3f72599d063a56cc0</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Diversity of Salmonella spp. serovars isolated...</td>\n",
       "      <td>10.1186/1746-6148-8-201</td>\n",
       "      <td>PMC3514206</td>\n",
       "      <td>23098237</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>BACKGROUND: Salmonellosis in water buffalo (Bu...</td>\n",
       "      <td>2012-10-25</td>\n",
       "      <td>...</td>\n",
       "      <td>BMC Vet Res</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/854e623d1f875e4605b2f...</td>\n",
       "      <td>document_parses/pmc_json/PMC3514206.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "      <td>Diversity of Salmonella spp. serovars isolated...</td>\n",
       "      <td>BACKGROUND: Salmonellosis in water buffalo (Bu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>z65xxk1f</td>\n",
       "      <td>3695704554777889f8232a9ea086df70bf17ff58</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Severe Childhood Malaria Syndromes Defined by ...</td>\n",
       "      <td>10.1371/journal.pone.0049778</td>\n",
       "      <td>PMC3514223</td>\n",
       "      <td>23226502</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>BACKGROUND: Cerebral malaria (CM) and severe m...</td>\n",
       "      <td>2012-12-04</td>\n",
       "      <td>...</td>\n",
       "      <td>PLoS One</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/3695704554777889f8232...</td>\n",
       "      <td>document_parses/pmc_json/PMC3514223.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "      <td>Severe Childhood Malaria Syndromes Defined by ...</td>\n",
       "      <td>BACKGROUND: Cerebral malaria (CM) and severe m...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>iuglkdcp</td>\n",
       "      <td>5a054a570f409d5c2006369f19addc98d024b6c1</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Predicting pseudoknotted structures across two...</td>\n",
       "      <td>10.1093/bioinformatics/bts575</td>\n",
       "      <td>PMC3516145</td>\n",
       "      <td>23044552</td>\n",
       "      <td>cc-by</td>\n",
       "      <td>Motivation: Laboratory RNA structure determina...</td>\n",
       "      <td>2012-12-01</td>\n",
       "      <td>...</td>\n",
       "      <td>Bioinformatics</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>document_parses/pdf_json/5a054a570f409d5c20063...</td>\n",
       "      <td>document_parses/pmc_json/PMC3516145.xml.json</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...</td>\n",
       "      <td>None</td>\n",
       "      <td>Predicting pseudoknotted structures across two...</td>\n",
       "      <td>Motivation: Laboratory RNA structure determina...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     cord_uid                                       sha source_x  \\\n",
       "0    ug7v899j  d1aafb70c066a2068b02786f8929fd9c900897fb      PMC   \n",
       "1    02tnwd4m  6b0567729c2143a66d737eb0a2f63f2dce2e5a7d      PMC   \n",
       "2    ejv2xln0  06ced00a5fc04215949aa72528f2eeaae1d58927      PMC   \n",
       "3    2b73a28n  348055649b6b8cf2b9a376498df9bf41f7123605      PMC   \n",
       "4    9785vg6d  5f48792a5fa08bed9f56016f4981ae2ca6031b32      PMC   \n",
       "..        ...                                       ...      ...   \n",
       "995  880nqc0f  ed2de3694f5580ea38f4adf24bd1c8b46862df1f      PMC   \n",
       "996  01b0vnnm  2a7c951e191425fd9fa5ac108f07a1f02eb75872      PMC   \n",
       "997  5b29wtim  854e623d1f875e4605b2ffd3f72599d063a56cc0      PMC   \n",
       "998  z65xxk1f  3695704554777889f8232a9ea086df70bf17ff58      PMC   \n",
       "999  iuglkdcp  5a054a570f409d5c2006369f19addc98d024b6c1      PMC   \n",
       "\n",
       "                                                 title  \\\n",
       "0    Clinical features of culture-proven Mycoplasma...   \n",
       "1    Nitric oxide: a pro-inflammatory mediator in l...   \n",
       "2      Surfactant protein-D and pulmonary host defense   \n",
       "3                 Role of endothelin-1 in lung disease   \n",
       "4    Gene expression in epithelial cells in respons...   \n",
       "..                                                 ...   \n",
       "995  Mannose-binding lectin deficiency and acute ex...   \n",
       "996  The changing phenotype of microglia from homeo...   \n",
       "997  Diversity of Salmonella spp. serovars isolated...   \n",
       "998  Severe Childhood Malaria Syndromes Defined by ...   \n",
       "999  Predicting pseudoknotted structures across two...   \n",
       "\n",
       "                               doi       pmcid pubmed_id license  \\\n",
       "0            10.1186/1471-2334-1-6    PMC35282  11472636   no-cc   \n",
       "1                     10.1186/rr14    PMC59543  11667967   no-cc   \n",
       "2                     10.1186/rr19    PMC59549  11667972   no-cc   \n",
       "3                     10.1186/rr44    PMC59574  11686871   no-cc   \n",
       "4                     10.1186/rr61    PMC59580  11686888   no-cc   \n",
       "..                             ...         ...       ...     ...   \n",
       "995            10.2147/copd.s33714  PMC3514010  23226013   no-cc   \n",
       "996          10.1186/2047-9158-1-9  PMC3514090  23210447   cc-by   \n",
       "997        10.1186/1746-6148-8-201  PMC3514206  23098237   cc-by   \n",
       "998   10.1371/journal.pone.0049778  PMC3514223  23226502   cc-by   \n",
       "999  10.1093/bioinformatics/bts575  PMC3516145  23044552   cc-by   \n",
       "\n",
       "                                              abstract publish_time  ...  \\\n",
       "0    OBJECTIVE: This retrospective chart review des...   2001-07-04  ...   \n",
       "1    Inflammatory diseases of the respiratory tract...   2000-08-15  ...   \n",
       "2    Surfactant protein-D (SP-D) participates in th...   2000-08-25  ...   \n",
       "3    Endothelin-1 (ET-1) is a 21 amino acid peptide...   2001-02-22  ...   \n",
       "4    Respiratory syncytial virus (RSV) and pneumoni...   2001-05-11  ...   \n",
       "..                                                 ...          ...  ...   \n",
       "995  BACKGROUND: Mannose-binding lectin is a collec...   2012-11-23  ...   \n",
       "996  It has been nearly a century since the early d...   2012-04-24  ...   \n",
       "997  BACKGROUND: Salmonellosis in water buffalo (Bu...   2012-10-25  ...   \n",
       "998  BACKGROUND: Cerebral malaria (CM) and severe m...   2012-12-04  ...   \n",
       "999  Motivation: Laboratory RNA structure determina...   2012-12-01  ...   \n",
       "\n",
       "                             journal mag_id who_covidence_id arxiv_id  \\\n",
       "0                     BMC Infect Dis   None             None     None   \n",
       "1                         Respir Res   None             None     None   \n",
       "2                         Respir Res   None             None     None   \n",
       "3                         Respir Res   None             None     None   \n",
       "4                         Respir Res   None             None     None   \n",
       "..                               ...    ...              ...      ...   \n",
       "995  Int J Chron Obstruct Pulmon Dis   None             None     None   \n",
       "996              Transl Neurodegener   None             None     None   \n",
       "997                      BMC Vet Res   None             None     None   \n",
       "998                         PLoS One   None             None     None   \n",
       "999                   Bioinformatics   None             None     None   \n",
       "\n",
       "                                        pdf_json_files  \\\n",
       "0    document_parses/pdf_json/d1aafb70c066a2068b027...   \n",
       "1    document_parses/pdf_json/6b0567729c2143a66d737...   \n",
       "2    document_parses/pdf_json/06ced00a5fc04215949aa...   \n",
       "3    document_parses/pdf_json/348055649b6b8cf2b9a37...   \n",
       "4    document_parses/pdf_json/5f48792a5fa08bed9f560...   \n",
       "..                                                 ...   \n",
       "995  document_parses/pdf_json/ed2de3694f5580ea38f4a...   \n",
       "996  document_parses/pdf_json/2a7c951e191425fd9fa5a...   \n",
       "997  document_parses/pdf_json/854e623d1f875e4605b2f...   \n",
       "998  document_parses/pdf_json/3695704554777889f8232...   \n",
       "999  document_parses/pdf_json/5a054a570f409d5c20063...   \n",
       "\n",
       "                                   pmc_json_files  \\\n",
       "0      document_parses/pmc_json/PMC35282.xml.json   \n",
       "1      document_parses/pmc_json/PMC59543.xml.json   \n",
       "2      document_parses/pmc_json/PMC59549.xml.json   \n",
       "3      document_parses/pmc_json/PMC59574.xml.json   \n",
       "4      document_parses/pmc_json/PMC59580.xml.json   \n",
       "..                                            ...   \n",
       "995  document_parses/pmc_json/PMC3514010.xml.json   \n",
       "996  document_parses/pmc_json/PMC3514090.xml.json   \n",
       "997  document_parses/pmc_json/PMC3514206.xml.json   \n",
       "998  document_parses/pmc_json/PMC3514223.xml.json   \n",
       "999  document_parses/pmc_json/PMC3516145.xml.json   \n",
       "\n",
       "                                                   url s2_id  \\\n",
       "0    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None   \n",
       "1    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  None   \n",
       "2    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  None   \n",
       "3    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  None   \n",
       "4    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...  None   \n",
       "..                                                 ...   ...   \n",
       "995  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None   \n",
       "996  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None   \n",
       "997  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None   \n",
       "998  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None   \n",
       "999  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3...  None   \n",
       "\n",
       "                                              pr_title  \\\n",
       "0    Clinical features of culture-proven Mycoplasma...   \n",
       "1    Nitric oxide: a pro-inflammatory mediator in l...   \n",
       "2      Surfactant protein-D and pulmonary host defense   \n",
       "3                 Role of endothelin-1 in lung disease   \n",
       "4    Gene expression in epithelial cells in respons...   \n",
       "..                                                 ...   \n",
       "995  Mannose-binding lectin deficiency and acute ex...   \n",
       "996  The changing phenotype of microglia from homeo...   \n",
       "997  Diversity of Salmonella spp. serovars isolated...   \n",
       "998  Severe Childhood Malaria Syndromes Defined by ...   \n",
       "999  Predicting pseudoknotted structures across two...   \n",
       "\n",
       "                                           pr_abstract  \n",
       "0    OBJECTIVE: This retrospective chart review des...  \n",
       "1    Inflammatory diseases of the respiratory tract...  \n",
       "2    Surfactant protein-D (SP-D) participates in th...  \n",
       "3    Endothelin-1 (ET-1) is a 21 amino acid peptide...  \n",
       "4    Respiratory syncytial virus (RSV) and pneumoni...  \n",
       "..                                                 ...  \n",
       "995  BACKGROUND: Mannose-binding lectin is a collec...  \n",
       "996  It has been nearly a century since the early d...  \n",
       "997  BACKGROUND: Salmonellosis in water buffalo (Bu...  \n",
       "998  BACKGROUND: Cerebral malaria (CM) and severe m...  \n",
       "999  Motivation: Laboratory RNA structure determina...  \n",
       "\n",
       "[1000 rows x 21 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating index - covid\n",
      "b'{\\n  \"name\" : \"DESKTOP-K0D65LT\",\\n  \"cluster_name\" : \"elasticsearch\",\\n  \"cluster_uuid\" : \"P8VwA_sWSl-ykHakwwPzXQ\",\\n  \"version\" : {\\n    \"number\" : \"7.10.2\",\\n    \"build_flavor\" : \"default\",\\n    \"build_type\" : \"zip\",\\n    \"build_hash\" : \"747e1cc71def077253878a59143c1f785afa92b9\",\\n    \"build_date\" : \"2021-01-13T00:42:12.435326Z\",\\n    \"build_snapshot\" : false,\\n    \"lucene_version\" : \"8.7.0\",\\n    \"minimum_wire_compatibility_version\" : \"6.8.0\",\\n    \"minimum_index_compatibility_version\" : \"6.0.0-beta1\"\\n  },\\n  \"tagline\" : \"You Know, for Search\"\\n}\\n'\n",
      "Indexing\n",
      "Data indexed.\n",
      "Sentence splitting, text tokenisation and normalisation\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Conda\\lib\\site-packages\\ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \"\"\"\n",
      "C:\\Conda\\lib\\site-packages\\ipykernel_launcher.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparison of Text Data after Sentence splitting, text tokenisation and normalisation step\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>pr_title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>pr_abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clinical features of culture-proven Mycoplasma...</td>\n",
       "      <td>[clinical features of culture proven mycoplasm...</td>\n",
       "      <td>OBJECTIVE: This retrospective chart review des...</td>\n",
       "      <td>[objective this retrospective chart review des...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n",
       "      <td>[nitric oxide a pro inflammatory mediator in l...</td>\n",
       "      <td>Inflammatory diseases of the respiratory tract...</td>\n",
       "      <td>[inflammatory diseases of the respiratory trac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Surfactant protein-D and pulmonary host defense</td>\n",
       "      <td>[surfactant protein d and pulmonary host defense]</td>\n",
       "      <td>Surfactant protein-D (SP-D) participates in th...</td>\n",
       "      <td>[surfactant protein d sp d participates in the...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Role of endothelin-1 in lung disease</td>\n",
       "      <td>[role of endothelin in lung disease]</td>\n",
       "      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n",
       "      <td>[endothelin et is a amino acid peptide with di...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene expression in epithelial cells in respons...</td>\n",
       "      <td>[gene expression in epithelial cells in respon...</td>\n",
       "      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n",
       "      <td>[respiratory syncytial virus rsv and pneumonia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>Mannose-binding lectin deficiency and acute ex...</td>\n",
       "      <td>[mannose binding lectin deficiency and acute e...</td>\n",
       "      <td>BACKGROUND: Mannose-binding lectin is a collec...</td>\n",
       "      <td>[background mannose binding lectin is a collec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>The changing phenotype of microglia from homeo...</td>\n",
       "      <td>[the changing phenotype of microglia from home...</td>\n",
       "      <td>It has been nearly a century since the early d...</td>\n",
       "      <td>[it has been nearly a century since the early ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>Diversity of Salmonella spp. serovars isolated...</td>\n",
       "      <td>[diversity of salmonella spp, serovars isolate...</td>\n",
       "      <td>BACKGROUND: Salmonellosis in water buffalo (Bu...</td>\n",
       "      <td>[background salmonellosis in water buffalo bub...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>Severe Childhood Malaria Syndromes Defined by ...</td>\n",
       "      <td>[severe childhood malaria syndromes defined by...</td>\n",
       "      <td>BACKGROUND: Cerebral malaria (CM) and severe m...</td>\n",
       "      <td>[background cerebral malaria cm and severe mal...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>Predicting pseudoknotted structures across two...</td>\n",
       "      <td>[predicting pseudoknotted structures across tw...</td>\n",
       "      <td>Motivation: Laboratory RNA structure determina...</td>\n",
       "      <td>[motivation laboratory rna structure determina...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 title  \\\n",
       "0    Clinical features of culture-proven Mycoplasma...   \n",
       "1    Nitric oxide: a pro-inflammatory mediator in l...   \n",
       "2      Surfactant protein-D and pulmonary host defense   \n",
       "3                 Role of endothelin-1 in lung disease   \n",
       "4    Gene expression in epithelial cells in respons...   \n",
       "..                                                 ...   \n",
       "995  Mannose-binding lectin deficiency and acute ex...   \n",
       "996  The changing phenotype of microglia from homeo...   \n",
       "997  Diversity of Salmonella spp. serovars isolated...   \n",
       "998  Severe Childhood Malaria Syndromes Defined by ...   \n",
       "999  Predicting pseudoknotted structures across two...   \n",
       "\n",
       "                                              pr_title  \\\n",
       "0    [clinical features of culture proven mycoplasm...   \n",
       "1    [nitric oxide a pro inflammatory mediator in l...   \n",
       "2    [surfactant protein d and pulmonary host defense]   \n",
       "3                 [role of endothelin in lung disease]   \n",
       "4    [gene expression in epithelial cells in respon...   \n",
       "..                                                 ...   \n",
       "995  [mannose binding lectin deficiency and acute e...   \n",
       "996  [the changing phenotype of microglia from home...   \n",
       "997  [diversity of salmonella spp, serovars isolate...   \n",
       "998  [severe childhood malaria syndromes defined by...   \n",
       "999  [predicting pseudoknotted structures across tw...   \n",
       "\n",
       "                                              abstract  \\\n",
       "0    OBJECTIVE: This retrospective chart review des...   \n",
       "1    Inflammatory diseases of the respiratory tract...   \n",
       "2    Surfactant protein-D (SP-D) participates in th...   \n",
       "3    Endothelin-1 (ET-1) is a 21 amino acid peptide...   \n",
       "4    Respiratory syncytial virus (RSV) and pneumoni...   \n",
       "..                                                 ...   \n",
       "995  BACKGROUND: Mannose-binding lectin is a collec...   \n",
       "996  It has been nearly a century since the early d...   \n",
       "997  BACKGROUND: Salmonellosis in water buffalo (Bu...   \n",
       "998  BACKGROUND: Cerebral malaria (CM) and severe m...   \n",
       "999  Motivation: Laboratory RNA structure determina...   \n",
       "\n",
       "                                           pr_abstract  \n",
       "0    [objective this retrospective chart review des...  \n",
       "1    [inflammatory diseases of the respiratory trac...  \n",
       "2    [surfactant protein d sp d participates in the...  \n",
       "3    [endothelin et is a amino acid peptide with di...  \n",
       "4    [respiratory syncytial virus rsv and pneumonia...  \n",
       "..                                                 ...  \n",
       "995  [background mannose binding lectin is a collec...  \n",
       "996  [it has been nearly a century since the early ...  \n",
       "997  [background salmonellosis in water buffalo bub...  \n",
       "998  [background cerebral malaria cm and severe mal...  \n",
       "999  [motivation laboratory rna structure determina...  \n",
       "\n",
       "[1000 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selecting keywords\n",
      "Text Data after removing of stop-words\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Conda\\lib\\site-packages\\ipykernel_launcher.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n",
      "C:\\Conda\\lib\\site-packages\\ipykernel_launcher.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  after removing the cwd from sys.path.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pr_title</th>\n",
       "      <th>pr_abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[clinical features culture proven mycoplasma p...</td>\n",
       "      <td>[objective retrospective chart review describe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[nitric oxide pro inflammatory mediator lung d...</td>\n",
       "      <td>[inflammatory diseases respiratory tract commo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[surfactant protein pulmonary host defense]</td>\n",
       "      <td>[surfactant protein participates innate respon...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[role endothelin lung disease]</td>\n",
       "      <td>[endothelin amino acid peptide diverse biologi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[gene expression epithelial cells response pne...</td>\n",
       "      <td>[respiratory syncytial virus rsv pneumonia vir...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>[mannose binding lectin deficiency acute exace...</td>\n",
       "      <td>[background mannose binding lectin collectin i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>[changing phenotype microglia homeostasis dise...</td>\n",
       "      <td>[nearly century since early description microg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>[diversity salmonella spp, serovars isolated i...</td>\n",
       "      <td>[background salmonellosis water buffalo bubalu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>[severe childhood malaria syndromes defined pl...</td>\n",
       "      <td>[background cerebral malaria severe malarial a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>[predicting pseudoknotted structures across tw...</td>\n",
       "      <td>[motivation laboratory rna structure determina...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              pr_title  \\\n",
       "0    [clinical features culture proven mycoplasma p...   \n",
       "1    [nitric oxide pro inflammatory mediator lung d...   \n",
       "2          [surfactant protein pulmonary host defense]   \n",
       "3                       [role endothelin lung disease]   \n",
       "4    [gene expression epithelial cells response pne...   \n",
       "..                                                 ...   \n",
       "995  [mannose binding lectin deficiency acute exace...   \n",
       "996  [changing phenotype microglia homeostasis dise...   \n",
       "997  [diversity salmonella spp, serovars isolated i...   \n",
       "998  [severe childhood malaria syndromes defined pl...   \n",
       "999  [predicting pseudoknotted structures across tw...   \n",
       "\n",
       "                                           pr_abstract  \n",
       "0    [objective retrospective chart review describe...  \n",
       "1    [inflammatory diseases respiratory tract commo...  \n",
       "2    [surfactant protein participates innate respon...  \n",
       "3    [endothelin amino acid peptide diverse biologi...  \n",
       "4    [respiratory syncytial virus rsv pneumonia vir...  \n",
       "..                                                 ...  \n",
       "995  [background mannose binding lectin collectin i...  \n",
       "996  [nearly century since early description microg...  \n",
       "997  [background salmonellosis water buffalo bubalu...  \n",
       "998  [background cerebral malaria severe malarial a...  \n",
       "999  [motivation laboratory rna structure determina...  \n",
       "\n",
       "[1000 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "133627\n",
      "Showing first 2000 top-freqent words in the corpus\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Word</th>\n",
       "      <th>Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>virus</td>\n",
       "      <td>985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>influenza</td>\n",
       "      <td>945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>infection</td>\n",
       "      <td>740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cells</td>\n",
       "      <td>711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>patients</td>\n",
       "      <td>587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>find</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>great</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>duration</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>initiation</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000</th>\n",
       "      <td>sirna</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Word  Frequency\n",
       "1          virus        985\n",
       "2      influenza        945\n",
       "3      infection        740\n",
       "4          cells        711\n",
       "5       patients        587\n",
       "...          ...        ...\n",
       "996         find         28\n",
       "997        great         28\n",
       "998     duration         28\n",
       "999   initiation         28\n",
       "1000       sirna         28\n",
       "\n",
       "[1000 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Showing bigrams in the corpus found by Pointwise Mutual Information method\n",
      "Applying frequency filter: a bigramm occurs more than 7 times\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>bigram</th>\n",
       "      <th>PMI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rift valley</td>\n",
       "      <td>13.857927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "      <td>13.705924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bronchoalveolar lavage</td>\n",
       "      <td>13.327412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>coronary artery</td>\n",
       "      <td>13.327412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>vesicular stomatitis</td>\n",
       "      <td>13.327412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863</th>\n",
       "      <td>cells cells</td>\n",
       "      <td>1.665403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>864</th>\n",
       "      <td>disease virus</td>\n",
       "      <td>1.581372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>865</th>\n",
       "      <td>infection results</td>\n",
       "      <td>1.548103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>866</th>\n",
       "      <td>pandemic virus</td>\n",
       "      <td>1.325173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>867</th>\n",
       "      <td>infection human</td>\n",
       "      <td>1.179704</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>868 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     bigram        PMI\n",
       "0               rift valley  13.857927\n",
       "1     endoplasmic reticulum  13.705924\n",
       "2    bronchoalveolar lavage  13.327412\n",
       "3           coronary artery  13.327412\n",
       "4      vesicular stomatitis  13.327412\n",
       "..                      ...        ...\n",
       "863             cells cells   1.665403\n",
       "864           disease virus   1.581372\n",
       "865       infection results   1.548103\n",
       "866          pandemic virus   1.325173\n",
       "867         infection human   1.179704\n",
       "\n",
       "[868 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Showing trigrams in the corpus found by Pointwise Mutual Information method\n",
      "Applying frequency filter: a trigramm occurs more than 5 times\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>trigram</th>\n",
       "      <th>PMI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>mov avg cusum</td>\n",
       "      <td>29.411848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cortex magnoliae officinalis</td>\n",
       "      <td>29.148813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>systemic lupus erythematosus</td>\n",
       "      <td>25.663387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>autoimmune hemolytic anemia</td>\n",
       "      <td>25.052889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>methicillin resistant staphylococcus</td>\n",
       "      <td>24.947180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>influenza virus replication</td>\n",
       "      <td>9.049636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>influenza virus infections</td>\n",
       "      <td>8.849042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>pandemic influenza infection</td>\n",
       "      <td>8.659061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>response influenza virus</td>\n",
       "      <td>8.532673</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>human influenza virus</td>\n",
       "      <td>7.425360</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>224 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  trigram        PMI\n",
       "0                           mov avg cusum  29.411848\n",
       "1            cortex magnoliae officinalis  29.148813\n",
       "2            systemic lupus erythematosus  25.663387\n",
       "3             autoimmune hemolytic anemia  25.052889\n",
       "4    methicillin resistant staphylococcus  24.947180\n",
       "..                                    ...        ...\n",
       "219           influenza virus replication   9.049636\n",
       "220            influenza virus infections   8.849042\n",
       "221          pandemic influenza infection   8.659061\n",
       "222              response influenza virus   8.532673\n",
       "223                 human influenza virus   7.425360\n",
       "\n",
       "[224 rows x 2 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting keywords from texts using Pointwise Mutual Information method and TextRank\n",
      "Text 0 - Done\n",
      "Text 1 - Done\n",
      "Text 2 - Done\n",
      "Text 3 - Done\n",
      "Text 4 - Done\n",
      "Text 5 - Done\n",
      "Text 6 - Done\n",
      "Text 7 - Done\n",
      "Text 8 - Done\n",
      "Text 9 - Done\n",
      "Text 10 - Done\n",
      "Text 11 - Done\n",
      "Text 12 - Done\n",
      "Text 13 - Done\n",
      "Text 15 - Done\n",
      "Text 16 - Done\n",
      "Text 17 - Done\n",
      "Text 18 - Done\n",
      "Text 19 - Done\n",
      "Text 20 - Done\n",
      "Text 21 - Done\n",
      "Text 22 - Done\n",
      "Text 23 - Done\n",
      "Text 24 - Done\n",
      "Text 25 - Done\n",
      "Text 26 - Done\n",
      "Text 27 - Done\n",
      "Text 28 - Done\n",
      "Text 29 - Done\n",
      "Text 30 - Done\n",
      "Text 31 - Done\n",
      "Text 32 - Done\n",
      "Text 33 - Done\n",
      "Text 34 - Done\n",
      "Text 35 - Done\n",
      "Text 36 - Done\n",
      "Text 37 - Done\n",
      "Text 38 - Done\n",
      "Text 39 - Done\n",
      "Text 40 - Done\n",
      "Text 41 - Done\n",
      "Text 42 - Done\n",
      "Text 43 - Done\n",
      "Text 44 - Done\n",
      "Text 45 - Done\n",
      "Text 46 - Done\n",
      "Text 47 - Done\n",
      "Text 48 - Done\n",
      "Text 49 - Done\n",
      "Text 50 - Done\n",
      "Text 57 - Done\n",
      "Text 58 - Done\n",
      "Text 59 - Done\n",
      "Text 60 - Done\n",
      "Text 61 - Done\n",
      "Text 62 - Done\n",
      "Text 64 - Done\n",
      "Text 65 - Done\n",
      "Text 66 - Done\n",
      "Text 67 - Done\n",
      "Text 68 - Done\n",
      "Text 69 - Done\n",
      "Text 70 - Done\n",
      "Text 71 - Done\n",
      "Text 72 - Done\n",
      "Text 73 - Done\n",
      "Text 74 - Done\n",
      "Text 75 - Done\n",
      "Text 76 - Done\n",
      "Text 77 - Done\n",
      "Text 78 - Done\n",
      "Text 79 - Done\n",
      "Text 80 - Done\n",
      "Text 81 - Done\n",
      "Text 82 - Done\n",
      "Text 83 - Done\n",
      "Text 84 - Done\n",
      "Text 85 - Done\n",
      "Text 86 - Done\n",
      "Text 87 - Done\n",
      "Text 88 - Done\n",
      "Text 89 - Done\n",
      "Text 90 - Done\n",
      "Text 91 - Done\n",
      "Text 92 - Done\n",
      "Text 93 - Done\n",
      "Text 94 - Done\n",
      "Text 95 - Done\n",
      "Text 96 - Done\n",
      "Text 97 - Done\n",
      "Text 98 - Done\n",
      "Text 99 - Done\n",
      "Text 100 - Done\n",
      "Text 101 - Done\n",
      "Text 102 - Done\n",
      "Text 103 - Done\n",
      "Text 104 - Done\n",
      "Text 105 - Done\n",
      "Text 106 - Done\n",
      "Text 107 - Done\n",
      "Text 108 - Done\n",
      "Text 109 - Done\n",
      "Text 110 - Done\n",
      "Text 111 - Done\n",
      "Text 112 - Done\n",
      "Text 113 - Done\n",
      "Text 114 - Done\n",
      "Text 115 - Done\n",
      "Text 116 - Done\n",
      "Text 117 - Done\n",
      "Text 118 - Done\n",
      "Text 119 - Done\n",
      "Text 120 - Done\n",
      "Text 121 - Done\n",
      "Text 122 - Done\n",
      "Text 123 - Done\n",
      "Text 124 - Done\n",
      "Text 125 - Done\n",
      "Text 126 - Done\n",
      "Text 127 - Done\n",
      "Text 128 - Done\n",
      "Text 129 - Done\n",
      "Text 130 - Done\n",
      "Text 131 - Done\n",
      "Text 132 - Done\n",
      "Text 133 - Done\n",
      "Text 134 - Done\n",
      "Text 135 - Done\n",
      "Text 136 - Done\n",
      "Text 137 - Done\n",
      "Text 138 - Done\n",
      "Text 139 - Done\n",
      "Text 140 - Done\n",
      "Text 141 - Done\n",
      "Text 142 - Done\n",
      "Text 143 - Done\n",
      "Text 144 - Done\n",
      "Text 145 - Done\n",
      "Text 146 - Done\n",
      "Text 147 - Done\n",
      "Text 148 - Done\n",
      "Text 149 - Done\n",
      "Text 150 - Done\n",
      "Text 151 - Done\n",
      "Text 152 - Done\n",
      "Text 153 - Done\n",
      "Text 154 - Done\n",
      "Text 155 - Done\n",
      "Text 156 - Done\n",
      "Text 157 - Done\n",
      "Text 158 - Done\n",
      "Text 159 - Done\n",
      "Text 160 - Done\n",
      "Text 161 - Done\n",
      "Text 162 - Done\n",
      "Text 163 - Done\n",
      "Text 164 - Done\n",
      "Text 165 - Done\n",
      "Text 166 - Done\n",
      "Text 167 - Done\n",
      "Text 168 - Done\n",
      "Text 169 - Done\n",
      "Text 170 - Done\n",
      "Text 171 - Done\n",
      "Text 172 - Done\n",
      "Text 173 - Done\n",
      "Text 174 - Done\n",
      "Text 175 - Done\n",
      "Text 176 - Done\n",
      "Text 177 - Done\n",
      "Text 178 - Done\n",
      "Text 179 - Done\n",
      "Text 180 - Done\n",
      "Text 181 - Done\n",
      "Text 182 - Done\n",
      "Text 183 - Done\n",
      "Text 184 - Done\n",
      "Text 185 - Done\n",
      "Text 186 - Done\n",
      "Text 187 - Done\n",
      "Text 188 - Done\n",
      "Text 189 - Done\n",
      "Text 190 - Done\n",
      "Text 191 - Done\n",
      "Text 192 - Done\n",
      "Text 194 - Done\n",
      "Text 195 - Done\n",
      "Text 196 - Done\n",
      "Text 197 - Done\n",
      "Text 198 - Done\n",
      "Text 199 - Done\n",
      "Text 200 - Done\n",
      "Text 201 - Done\n",
      "Text 202 - Done\n",
      "Text 203 - Done\n",
      "Text 204 - Done\n",
      "Text 205 - Done\n",
      "Text 206 - Done\n",
      "Text 208 - Done\n",
      "Text 209 - Done\n",
      "Text 210 - Done\n",
      "Text 211 - Done\n",
      "Text 212 - Done\n",
      "Text 213 - Done\n",
      "Text 214 - Done\n",
      "Text 215 - Done\n",
      "Text 216 - Done\n",
      "Text 217 - Done\n",
      "Text 219 - Done\n",
      "Text 220 - Done\n",
      "Text 221 - Done\n",
      "Text 222 - Done\n",
      "Text 223 - Done\n",
      "Text 224 - Done\n",
      "Text 225 - Done\n",
      "Text 226 - Done\n",
      "Text 227 - Done\n",
      "Text 228 - Done\n",
      "Text 229 - Done\n",
      "Text 230 - Done\n",
      "Text 231 - Done\n",
      "Text 232 - Done\n",
      "Text 233 - Done\n",
      "Text 236 - Done\n",
      "Text 237 - Done\n",
      "Text 238 - Done\n",
      "Text 239 - Done\n",
      "Text 240 - Done\n",
      "Text 241 - Done\n",
      "Text 242 - Done\n",
      "Text 243 - Done\n",
      "Text 244 - Done\n",
      "Text 245 - Done\n",
      "Text 246 - Done\n",
      "Text 247 - Done\n",
      "Text 248 - Done\n",
      "Text 249 - Done\n",
      "Text 250 - Done\n",
      "Text 251 - Done\n",
      "Text 252 - Done\n",
      "Text 253 - Done\n",
      "Text 254 - Done\n",
      "Text 255 - Done\n",
      "Text 256 - Done\n",
      "Text 257 - Done\n",
      "Text 258 - Done\n",
      "Text 259 - Done\n",
      "Text 261 - Done\n",
      "Text 262 - Done\n",
      "Text 263 - Done\n",
      "Text 264 - Done\n",
      "Text 265 - Done\n",
      "Text 266 - Done\n",
      "Text 267 - Done\n",
      "Text 268 - Done\n",
      "Text 269 - Done\n",
      "Text 270 - Done\n",
      "Text 271 - Done\n",
      "Text 272 - Done\n",
      "Text 273 - Done\n",
      "Text 274 - Done\n",
      "Text 275 - Done\n",
      "Text 276 - Done\n",
      "Text 277 - Done\n",
      "Text 278 - Done\n",
      "Text 279 - Done\n",
      "Text 280 - Done\n",
      "Text 281 - Done\n",
      "Text 282 - Done\n",
      "Text 283 - Done\n",
      "Text 284 - Done\n",
      "Text 285 - Done\n",
      "Text 286 - Done\n",
      "Text 287 - Done\n",
      "Text 288 - Done\n",
      "Text 289 - Done\n",
      "Text 290 - Done\n",
      "Text 291 - Done\n",
      "Text 292 - Done\n",
      "Text 293 - Done\n",
      "Text 294 - Done\n",
      "Text 295 - Done\n",
      "Text 296 - Done\n",
      "Text 297 - Done\n",
      "Text 298 - Done\n",
      "Text 299 - Done\n",
      "Text 300 - Done\n",
      "Text 301 - Done\n",
      "Text 302 - Done\n",
      "Text 303 - Done\n",
      "Text 305 - Done\n",
      "Text 306 - Done\n",
      "Text 308 - Done\n",
      "Text 309 - Done\n",
      "Text 310 - Done\n",
      "Text 311 - Done\n",
      "Text 312 - Done\n",
      "Text 313 - Done\n",
      "Text 314 - Done\n",
      "Text 315 - Done\n",
      "Text 316 - Done\n",
      "Text 317 - Done\n",
      "Text 318 - Done\n",
      "Text 320 - Done\n",
      "Text 321 - Done\n",
      "Text 322 - Done\n",
      "Text 323 - Done\n",
      "Text 324 - Done\n",
      "Text 325 - Done\n",
      "Text 326 - Done\n",
      "Text 327 - Done\n",
      "Text 328 - Done\n",
      "Text 329 - Done\n",
      "Text 330 - Done\n",
      "Text 331 - Done\n",
      "Text 332 - Done\n",
      "Text 333 - Done\n",
      "Text 334 - Done\n",
      "Text 335 - Done\n",
      "Text 336 - Done\n",
      "Text 337 - Done\n",
      "Text 338 - Done\n",
      "Text 339 - Done\n",
      "Text 341 - Done\n",
      "Text 342 - Done\n",
      "Text 343 - Done\n",
      "Text 344 - Done\n",
      "Text 345 - Done\n",
      "Text 346 - Done\n",
      "Text 347 - Done\n",
      "Text 348 - Done\n",
      "Text 349 - Done\n",
      "Text 350 - Done\n",
      "Text 351 - Done\n",
      "Text 352 - Done\n",
      "Text 353 - Done\n",
      "Text 354 - Done\n",
      "Text 355 - Done\n",
      "Text 356 - Done\n",
      "Text 357 - Done\n",
      "Text 358 - Done\n",
      "Text 359 - Done\n",
      "Text 360 - Done\n",
      "Text 361 - Done\n",
      "Text 362 - Done\n",
      "Text 363 - Done\n",
      "Text 364 - Done\n",
      "Text 365 - Done\n",
      "Text 366 - Done\n",
      "Text 367 - Done\n",
      "Text 368 - Done\n",
      "Text 369 - Done\n",
      "Text 370 - Done\n",
      "Text 371 - Done\n",
      "Text 372 - Done\n",
      "Text 373 - Done\n",
      "Text 374 - Done\n",
      "Text 376 - Done\n",
      "Text 377 - Done\n",
      "Text 378 - Done\n",
      "Text 379 - Done\n",
      "Text 380 - Done\n",
      "Text 381 - Done\n",
      "Text 382 - Done\n",
      "Text 383 - Done\n",
      "Text 384 - Done\n",
      "Text 385 - Done\n",
      "Text 386 - Done\n",
      "Text 387 - Done\n",
      "Text 388 - Done\n",
      "Text 389 - Done\n",
      "Text 390 - Done\n",
      "Text 391 - Done\n",
      "Text 392 - Done\n",
      "Text 393 - Done\n",
      "Text 394 - Done\n",
      "Text 396 - Done\n",
      "Text 397 - Done\n",
      "Text 398 - Done\n",
      "Text 399 - Done\n",
      "Text 400 - Done\n",
      "Text 401 - Done\n",
      "Text 402 - Done\n",
      "Text 403 - Done\n",
      "Text 404 - Done\n",
      "Text 405 - Done\n",
      "Text 406 - Done\n",
      "Text 407 - Done\n",
      "Text 408 - Done\n",
      "Text 409 - Done\n",
      "Text 410 - Done\n",
      "Text 411 - Done\n",
      "Text 412 - Done\n",
      "Text 413 - Done\n",
      "Text 414 - Done\n",
      "Text 415 - Done\n",
      "Text 416 - Done\n",
      "Text 417 - Done\n",
      "Text 418 - Done\n",
      "Text 419 - Done\n",
      "Text 421 - Done\n",
      "Text 422 - Done\n",
      "Text 423 - Done\n",
      "Text 424 - Done\n",
      "Text 425 - Done\n",
      "Text 426 - Done\n",
      "Text 427 - Done\n",
      "Text 428 - Done\n",
      "Text 429 - Done\n",
      "Text 430 - Done\n",
      "Text 431 - Done\n",
      "Text 432 - Done\n",
      "Text 433 - Done\n",
      "Text 434 - Done\n",
      "Text 435 - Done\n",
      "Text 436 - Done\n",
      "Text 437 - Done\n",
      "Text 438 - Done\n",
      "Text 439 - Done\n",
      "Text 440 - Done\n",
      "Text 441 - Done\n",
      "Text 442 - Done\n",
      "Text 443 - Done\n",
      "Text 444 - Done\n",
      "Text 445 - Done\n",
      "Text 446 - Done\n",
      "Text 447 - Done\n",
      "Text 448 - Done\n",
      "Text 449 - Done\n",
      "Text 450 - Done\n",
      "Text 451 - Done\n",
      "Text 452 - Done\n",
      "Text 453 - Done\n",
      "Text 454 - Done\n",
      "Text 455 - Done\n",
      "Text 456 - Done\n",
      "Text 457 - Done\n",
      "Text 458 - Done\n",
      "Text 459 - Done\n",
      "Text 460 - Done\n",
      "Text 461 - Done\n",
      "Text 462 - Done\n",
      "Text 463 - Done\n",
      "Text 464 - Done\n",
      "Text 465 - Done\n",
      "Text 466 - Done\n",
      "Text 467 - Done\n",
      "Text 468 - Done\n",
      "Text 469 - Done\n",
      "Text 470 - Done\n",
      "Text 471 - Done\n",
      "Text 472 - Done\n",
      "Text 473 - Done\n",
      "Text 474 - Done\n",
      "Text 475 - Done\n",
      "Text 476 - Done\n",
      "Text 477 - Done\n",
      "Text 478 - Done\n",
      "Text 479 - Done\n",
      "Text 480 - Done\n",
      "Text 481 - Done\n",
      "Text 482 - Done\n",
      "Text 483 - Done\n",
      "Text 484 - Done\n",
      "Text 485 - Done\n",
      "Text 486 - Done\n",
      "Text 488 - Done\n",
      "Text 489 - Done\n",
      "Text 490 - Done\n",
      "Text 491 - Done\n",
      "Text 492 - Done\n",
      "Text 493 - Done\n",
      "Text 494 - Done\n",
      "Text 495 - Done\n",
      "Text 496 - Done\n",
      "Text 497 - Done\n",
      "Text 498 - Done\n",
      "Text 499 - Done\n",
      "Text 500 - Done\n",
      "Text 501 - Done\n",
      "Text 502 - Done\n",
      "Text 503 - Done\n",
      "Text 504 - Done\n",
      "Text 505 - Done\n",
      "Text 506 - Done\n",
      "Text 507 - Done\n",
      "Text 508 - Done\n",
      "Text 509 - Done\n",
      "Text 510 - Done\n",
      "Text 511 - Done\n",
      "Text 512 - Done\n",
      "Text 513 - Done\n",
      "Text 514 - Done\n",
      "Text 515 - Done\n",
      "Text 516 - Done\n",
      "Text 517 - Done\n",
      "Text 518 - Done\n",
      "Text 519 - Done\n",
      "Text 520 - Done\n",
      "Text 521 - Done\n",
      "Text 523 - Done\n",
      "Text 524 - Done\n",
      "Text 525 - Done\n",
      "Text 526 - Done\n",
      "Text 527 - Done\n",
      "Text 528 - Done\n",
      "Text 529 - Done\n",
      "Text 530 - Done\n",
      "Text 531 - Done\n",
      "Text 532 - Done\n",
      "Text 533 - Done\n",
      "Text 534 - Done\n",
      "Text 535 - Done\n",
      "Text 536 - Done\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Text 537 - Done\n",
      "Text 538 - Done\n",
      "Text 539 - Done\n",
      "Text 540 - Done\n",
      "Text 541 - Done\n",
      "Text 542 - Done\n",
      "Text 543 - Done\n",
      "Text 544 - Done\n",
      "Text 545 - Done\n",
      "Text 546 - Done\n",
      "Text 547 - Done\n",
      "Text 548 - Done\n",
      "Text 549 - Done\n",
      "Text 550 - Done\n",
      "Text 551 - Done\n",
      "Text 552 - Done\n",
      "Text 553 - Done\n",
      "Text 554 - Done\n",
      "Text 555 - Done\n",
      "Text 556 - Done\n",
      "Text 557 - Done\n",
      "Text 558 - Done\n",
      "Text 559 - Done\n",
      "Text 560 - Done\n",
      "Text 561 - Done\n",
      "Text 562 - Done\n",
      "Text 563 - Done\n",
      "Text 564 - Done\n",
      "Text 566 - Done\n",
      "Text 567 - Done\n",
      "Text 568 - Done\n",
      "Text 569 - Done\n",
      "Text 571 - Done\n",
      "Text 572 - Done\n",
      "Text 573 - Done\n",
      "Text 574 - Done\n",
      "Text 575 - Done\n",
      "Text 576 - Done\n",
      "Text 577 - Done\n",
      "Text 578 - Done\n",
      "Text 579 - Done\n",
      "Text 580 - Done\n",
      "Text 581 - Done\n",
      "Text 582 - Done\n",
      "Text 583 - Done\n",
      "Text 584 - Done\n",
      "Text 585 - Done\n",
      "Text 586 - Done\n",
      "Text 587 - Done\n",
      "Text 588 - Done\n",
      "Text 589 - Done\n",
      "Text 590 - Done\n",
      "Text 591 - Done\n",
      "Text 592 - Done\n",
      "Text 593 - Done\n",
      "Text 594 - Done\n",
      "Text 595 - Done\n",
      "Text 596 - Done\n",
      "Text 597 - Done\n",
      "Text 598 - Done\n",
      "Text 599 - Done\n",
      "Text 600 - Done\n",
      "Text 601 - Done\n",
      "Text 602 - Done\n",
      "Text 603 - Done\n",
      "Text 604 - Done\n",
      "Text 605 - Done\n",
      "Text 606 - Done\n",
      "Text 607 - Done\n",
      "Text 608 - Done\n",
      "Text 609 - Done\n",
      "Text 610 - Done\n",
      "Text 611 - Done\n",
      "Text 612 - Done\n",
      "Text 613 - Done\n",
      "Text 614 - Done\n",
      "Text 615 - Done\n",
      "Text 617 - Done\n",
      "Text 618 - Done\n",
      "Text 619 - Done\n",
      "Text 620 - Done\n",
      "Text 621 - Done\n",
      "Text 622 - Done\n",
      "Text 623 - Done\n",
      "Text 624 - Done\n",
      "Text 625 - Done\n",
      "Text 626 - Done\n",
      "Text 627 - Done\n",
      "Text 628 - Done\n",
      "Text 629 - Done\n",
      "Text 630 - Done\n",
      "Text 631 - Done\n",
      "Text 632 - Done\n",
      "Text 633 - Done\n",
      "Text 634 - Done\n",
      "Text 635 - Done\n",
      "Text 636 - Done\n",
      "Text 637 - Done\n",
      "Text 638 - Done\n",
      "Text 639 - Done\n",
      "Text 640 - Done\n",
      "Text 641 - Done\n",
      "Text 642 - Done\n",
      "Text 643 - Done\n",
      "Text 644 - Done\n",
      "Text 645 - Done\n",
      "Text 646 - Done\n",
      "Text 647 - Done\n",
      "Text 648 - Done\n",
      "Text 649 - Done\n",
      "Text 650 - Done\n",
      "Text 651 - Done\n",
      "Text 652 - Done\n",
      "Text 653 - Done\n",
      "Text 654 - Done\n",
      "Text 655 - Done\n",
      "Text 656 - Done\n",
      "Text 657 - Done\n",
      "Text 658 - Done\n",
      "Text 659 - Done\n",
      "Text 660 - Done\n",
      "Text 661 - Done\n",
      "Text 662 - Done\n",
      "Text 663 - Done\n",
      "Text 664 - Done\n",
      "Text 665 - Done\n",
      "Text 666 - Done\n",
      "Text 667 - Done\n",
      "Text 668 - Done\n",
      "Text 669 - Done\n",
      "Text 670 - Done\n",
      "Text 672 - Done\n",
      "Text 674 - Done\n",
      "Text 675 - Done\n",
      "Text 676 - Done\n",
      "Text 677 - Done\n",
      "Text 678 - Done\n",
      "Text 679 - Done\n",
      "Text 680 - Done\n",
      "Text 681 - Done\n",
      "Text 682 - Done\n",
      "Text 683 - Done\n",
      "Text 684 - Done\n",
      "Text 685 - Done\n",
      "Text 686 - Done\n",
      "Text 687 - Done\n",
      "Text 688 - Done\n",
      "Text 689 - Done\n",
      "Text 690 - Done\n",
      "Text 691 - Done\n",
      "Text 692 - Done\n",
      "Text 693 - Done\n",
      "Text 694 - Done\n",
      "Text 695 - Done\n",
      "Text 696 - Done\n",
      "Text 697 - Done\n",
      "Text 698 - Done\n",
      "Text 699 - Done\n",
      "Text 700 - Done\n",
      "Text 701 - Done\n",
      "Text 702 - Done\n",
      "Text 703 - Done\n",
      "Text 704 - Done\n",
      "Text 705 - Done\n",
      "Text 706 - Done\n",
      "Text 707 - Done\n",
      "Text 708 - Done\n",
      "Text 709 - Done\n",
      "Text 710 - Done\n",
      "Text 711 - Done\n",
      "Text 712 - Done\n",
      "Text 713 - Done\n",
      "Text 714 - Done\n",
      "Text 715 - Done\n",
      "Text 716 - Done\n",
      "Text 717 - Done\n",
      "Text 718 - Done\n",
      "Text 719 - Done\n",
      "Text 720 - Done\n",
      "Text 721 - Done\n",
      "Text 722 - Done\n",
      "Text 723 - Done\n",
      "Text 724 - Done\n",
      "Text 725 - Done\n",
      "Text 726 - Done\n",
      "Text 727 - Done\n",
      "Text 728 - Done\n",
      "Text 729 - Done\n",
      "Text 730 - Done\n",
      "Text 731 - Done\n",
      "Text 733 - Done\n",
      "Text 734 - Done\n",
      "Text 735 - Done\n",
      "Text 736 - Done\n",
      "Text 737 - Done\n",
      "Text 738 - Done\n",
      "Text 739 - Done\n",
      "Text 740 - Done\n",
      "Text 741 - Done\n",
      "Text 742 - Done\n",
      "Text 743 - Done\n",
      "Text 744 - Done\n",
      "Text 745 - Done\n",
      "Text 746 - Done\n",
      "Text 747 - Done\n",
      "Text 748 - Done\n",
      "Text 749 - Done\n",
      "Text 750 - Done\n",
      "Text 751 - Done\n",
      "Text 752 - Done\n",
      "Text 753 - Done\n",
      "Text 754 - Done\n",
      "Text 755 - Done\n",
      "Text 756 - Done\n",
      "Text 757 - Done\n",
      "Text 758 - Done\n",
      "Text 762 - Done\n",
      "Text 763 - Done\n",
      "Text 764 - Done\n",
      "Text 765 - Done\n",
      "Text 766 - Done\n",
      "Text 767 - Done\n",
      "Text 768 - Done\n",
      "Text 770 - Done\n",
      "Text 771 - Done\n",
      "Text 772 - Done\n",
      "Text 773 - Done\n",
      "Text 774 - Done\n",
      "Text 776 - Done\n",
      "Text 777 - Done\n",
      "Text 778 - Done\n",
      "Text 779 - Done\n",
      "Text 780 - Done\n",
      "Text 781 - Done\n",
      "Text 782 - Done\n",
      "Text 783 - Done\n",
      "Text 784 - Done\n",
      "Text 785 - Done\n",
      "Text 786 - Done\n",
      "Text 787 - Done\n",
      "Text 788 - Done\n",
      "Text 789 - Done\n",
      "Text 790 - Done\n",
      "Text 791 - Done\n",
      "Text 792 - Done\n",
      "Text 793 - Done\n",
      "Text 794 - Done\n",
      "Text 795 - Done\n",
      "Text 796 - Done\n",
      "Text 797 - Done\n",
      "Text 798 - Done\n",
      "Text 799 - Done\n",
      "Text 800 - Done\n",
      "Text 801 - Done\n",
      "Text 802 - Done\n",
      "Text 803 - Done\n",
      "Text 804 - Done\n",
      "Text 805 - Done\n",
      "Text 806 - Done\n",
      "Text 807 - Done\n",
      "Text 811 - Done\n",
      "Text 812 - Done\n",
      "Text 813 - Done\n",
      "Text 814 - Done\n",
      "Text 815 - Done\n",
      "Text 816 - Done\n",
      "Text 817 - Done\n",
      "Text 818 - Done\n",
      "Text 819 - Done\n",
      "Text 820 - Done\n",
      "Text 821 - Done\n",
      "Text 822 - Done\n",
      "Text 823 - Done\n",
      "Text 824 - Done\n",
      "Text 825 - Done\n",
      "Text 826 - Done\n",
      "Text 827 - Done\n",
      "Text 828 - Done\n",
      "Text 829 - Done\n",
      "Text 830 - Done\n",
      "Text 831 - Done\n",
      "Text 832 - Done\n",
      "Text 833 - Done\n",
      "Text 834 - Done\n",
      "Text 835 - Done\n",
      "Text 836 - Done\n",
      "Text 837 - Done\n",
      "Text 838 - Done\n",
      "Text 839 - Done\n",
      "Text 840 - Done\n",
      "Text 841 - Done\n",
      "Text 842 - Done\n",
      "Text 843 - Done\n",
      "Text 845 - Done\n",
      "Text 846 - Done\n",
      "Text 847 - Done\n",
      "Text 849 - Done\n",
      "Text 850 - Done\n",
      "Text 851 - Done\n",
      "Text 852 - Done\n",
      "Text 853 - Done\n",
      "Text 854 - Done\n",
      "Text 855 - Done\n",
      "Text 856 - Done\n",
      "Text 857 - Done\n",
      "Text 858 - Done\n",
      "Text 859 - Done\n",
      "Text 860 - Done\n",
      "Text 861 - Done\n",
      "Text 862 - Done\n",
      "Text 863 - Done\n",
      "Text 864 - Done\n",
      "Text 866 - Done\n",
      "Text 867 - Done\n",
      "Text 869 - Done\n",
      "Text 870 - Done\n",
      "Text 871 - Done\n",
      "Text 872 - Done\n",
      "Text 873 - Done\n",
      "Text 874 - Done\n",
      "Text 875 - Done\n",
      "Text 877 - Done\n",
      "Text 878 - Done\n",
      "Text 879 - Done\n",
      "Text 880 - Done\n",
      "Text 881 - Done\n",
      "Text 882 - Done\n",
      "Text 883 - Done\n",
      "Text 884 - Done\n",
      "Text 885 - Done\n",
      "Text 886 - Done\n",
      "Text 887 - Done\n",
      "Text 888 - Done\n",
      "Text 889 - Done\n",
      "Text 890 - Done\n",
      "Text 891 - Done\n",
      "Text 892 - Done\n",
      "Text 893 - Done\n",
      "Text 894 - Done\n",
      "Text 896 - Done\n",
      "Text 897 - Done\n",
      "Text 898 - Done\n",
      "Text 899 - Done\n",
      "Text 900 - Done\n",
      "Text 901 - Done\n",
      "Text 902 - Done\n",
      "Text 903 - Done\n",
      "Text 904 - Done\n",
      "Text 905 - Done\n",
      "Text 906 - Done\n",
      "Text 907 - Done\n",
      "Text 908 - Done\n",
      "Text 909 - Done\n",
      "Text 910 - Done\n",
      "Text 911 - Done\n",
      "Text 912 - Done\n",
      "Text 913 - Done\n",
      "Text 914 - Done\n",
      "Text 915 - Done\n",
      "Text 916 - Done\n",
      "Text 917 - Done\n",
      "Text 918 - Done\n",
      "Text 919 - Done\n",
      "Text 920 - Done\n",
      "Text 921 - Done\n",
      "Text 922 - Done\n",
      "Text 923 - Done\n",
      "Text 924 - Done\n",
      "Text 925 - Done\n",
      "Text 926 - Done\n",
      "Text 927 - Done\n",
      "Text 928 - Done\n",
      "Text 929 - Done\n",
      "Text 930 - Done\n",
      "Text 931 - Done\n",
      "Text 932 - Done\n",
      "Text 933 - Done\n",
      "Text 934 - Done\n",
      "Text 935 - Done\n",
      "Text 936 - Done\n",
      "Text 937 - Done\n",
      "Text 938 - Done\n",
      "Text 939 - Done\n",
      "Text 940 - Done\n",
      "Text 941 - Done\n",
      "Text 942 - Done\n",
      "Text 943 - Done\n",
      "Text 944 - Done\n",
      "Text 945 - Done\n",
      "Text 946 - Done\n",
      "Text 947 - Done\n",
      "Text 948 - Done\n",
      "Text 949 - Done\n",
      "Text 950 - Done\n",
      "Text 951 - Done\n",
      "Text 952 - Done\n",
      "Text 953 - Done\n",
      "Text 954 - Done\n",
      "Text 955 - Done\n",
      "Text 956 - Done\n",
      "Text 957 - Done\n",
      "Text 958 - Done\n",
      "Text 959 - Done\n",
      "Text 960 - Done\n",
      "Text 961 - Done\n",
      "Text 962 - Done\n",
      "Text 963 - Done\n",
      "Text 964 - Done\n",
      "Text 965 - Done\n",
      "Text 966 - Done\n",
      "Text 968 - Done\n",
      "Text 969 - Done\n",
      "Text 970 - Done\n",
      "Text 971 - Done\n",
      "Text 972 - Done\n",
      "Text 973 - Done\n",
      "Text 974 - Done\n",
      "Text 975 - Done\n",
      "Text 976 - Done\n",
      "Text 977 - Done\n",
      "Text 978 - Done\n",
      "Text 979 - Done\n",
      "Text 980 - Done\n",
      "Text 981 - Done\n",
      "Text 982 - Done\n",
      "Text 983 - Done\n",
      "Text 984 - Done\n",
      "Text 985 - Done\n",
      "Text 986 - Done\n",
      "Text 987 - Done\n",
      "Text 988 - Done\n",
      "Text 989 - Done\n",
      "Text 990 - Done\n",
      "Text 991 - Done\n",
      "Text 992 - Done\n",
      "Text 993 - Done\n",
      "Text 994 - Done\n",
      "Text 995 - Done\n",
      "Text 996 - Done\n",
      "Text 997 - Done\n",
      "Text 998 - Done\n",
      "Text 999 - Done\n",
      "1868\n",
      "19754\n",
      "19754\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Conda\\lib\\site-packages\\ipykernel_launcher.py:78: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Comparison of Text Data after Keywords Extraction using Pointwise Mutual Information method and TextRank\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>pr_title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>pr_abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clinical features of culture-proven Mycoplasma...</td>\n",
       "      <td>[clinical features culture proven mycoplasma p...</td>\n",
       "      <td>OBJECTIVE: This retrospective chart review des...</td>\n",
       "      <td>[objective_retrospective_chart_review epidemio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n",
       "      <td>[nitric oxide pro inflammatory mediator lung d...</td>\n",
       "      <td>Inflammatory diseases of the respiratory tract...</td>\n",
       "      <td>[inflammatory_diseases respiratory_tract commo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Surfactant protein-D and pulmonary host defense</td>\n",
       "      <td>[surfactant protein pulmonary host defense]</td>\n",
       "      <td>Surfactant protein-D (SP-D) participates in th...</td>\n",
       "      <td>[surfactant_protein_participates response micr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Role of endothelin-1 in lung disease</td>\n",
       "      <td>[role endothelin lung disease]</td>\n",
       "      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n",
       "      <td>[endothelin_amino_acid_peptide_diverse_biologi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene expression in epithelial cells in respons...</td>\n",
       "      <td>[gene expression epithelial cells response pne...</td>\n",
       "      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n",
       "      <td>[respiratory_syncytial_virus_rsv_pneumonia_vir...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>Mannose-binding lectin deficiency and acute ex...</td>\n",
       "      <td>[mannose binding lectin deficiency acute exace...</td>\n",
       "      <td>BACKGROUND: Mannose-binding lectin is a collec...</td>\n",
       "      <td>[background_mannose_binding_lectin_collectin h...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>The changing phenotype of microglia from homeo...</td>\n",
       "      <td>[changing phenotype microglia homeostasis dise...</td>\n",
       "      <td>It has been nearly a century since the early d...</td>\n",
       "      <td>[century early_description_microglia_rio many_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>Diversity of Salmonella spp. serovars isolated...</td>\n",
       "      <td>[diversity salmonella spp, serovars isolated i...</td>\n",
       "      <td>BACKGROUND: Salmonellosis in water buffalo (Bu...</td>\n",
       "      <td>[background_salmonellosis_water_buffalo_bubalu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>Severe Childhood Malaria Syndromes Defined by ...</td>\n",
       "      <td>[severe childhood malaria syndromes defined pl...</td>\n",
       "      <td>BACKGROUND: Cerebral malaria (CM) and severe m...</td>\n",
       "      <td>[background_cerebral_malaria serious_life clin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>Predicting pseudoknotted structures across two...</td>\n",
       "      <td>[predicting pseudoknotted structures across tw...</td>\n",
       "      <td>Motivation: Laboratory RNA structure determina...</td>\n",
       "      <td>[motivation_laboratory_rna_structure_determina...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 title  \\\n",
       "0    Clinical features of culture-proven Mycoplasma...   \n",
       "1    Nitric oxide: a pro-inflammatory mediator in l...   \n",
       "2      Surfactant protein-D and pulmonary host defense   \n",
       "3                 Role of endothelin-1 in lung disease   \n",
       "4    Gene expression in epithelial cells in respons...   \n",
       "..                                                 ...   \n",
       "995  Mannose-binding lectin deficiency and acute ex...   \n",
       "996  The changing phenotype of microglia from homeo...   \n",
       "997  Diversity of Salmonella spp. serovars isolated...   \n",
       "998  Severe Childhood Malaria Syndromes Defined by ...   \n",
       "999  Predicting pseudoknotted structures across two...   \n",
       "\n",
       "                                              pr_title  \\\n",
       "0    [clinical features culture proven mycoplasma p...   \n",
       "1    [nitric oxide pro inflammatory mediator lung d...   \n",
       "2          [surfactant protein pulmonary host defense]   \n",
       "3                       [role endothelin lung disease]   \n",
       "4    [gene expression epithelial cells response pne...   \n",
       "..                                                 ...   \n",
       "995  [mannose binding lectin deficiency acute exace...   \n",
       "996  [changing phenotype microglia homeostasis dise...   \n",
       "997  [diversity salmonella spp, serovars isolated i...   \n",
       "998  [severe childhood malaria syndromes defined pl...   \n",
       "999  [predicting pseudoknotted structures across tw...   \n",
       "\n",
       "                                              abstract  \\\n",
       "0    OBJECTIVE: This retrospective chart review des...   \n",
       "1    Inflammatory diseases of the respiratory tract...   \n",
       "2    Surfactant protein-D (SP-D) participates in th...   \n",
       "3    Endothelin-1 (ET-1) is a 21 amino acid peptide...   \n",
       "4    Respiratory syncytial virus (RSV) and pneumoni...   \n",
       "..                                                 ...   \n",
       "995  BACKGROUND: Mannose-binding lectin is a collec...   \n",
       "996  It has been nearly a century since the early d...   \n",
       "997  BACKGROUND: Salmonellosis in water buffalo (Bu...   \n",
       "998  BACKGROUND: Cerebral malaria (CM) and severe m...   \n",
       "999  Motivation: Laboratory RNA structure determina...   \n",
       "\n",
       "                                           pr_abstract  \n",
       "0    [objective_retrospective_chart_review epidemio...  \n",
       "1    [inflammatory_diseases respiratory_tract commo...  \n",
       "2    [surfactant_protein_participates response micr...  \n",
       "3    [endothelin_amino_acid_peptide_diverse_biologi...  \n",
       "4    [respiratory_syncytial_virus_rsv_pneumonia_vir...  \n",
       "..                                                 ...  \n",
       "995  [background_mannose_binding_lectin_collectin h...  \n",
       "996  [century early_description_microglia_rio many_...  \n",
       "997  [background_salmonellosis_water_buffalo_bubalu...  \n",
       "998  [background_cerebral_malaria serious_life clin...  \n",
       "999  [motivation_laboratory_rna_structure_determina...  \n",
       "\n",
       "[1000 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting keywords from texts using TF/IDF\n",
      "Comparison of Text Data after Selecting keywords step\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>pr_title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>pr_abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clinical features of culture-proven Mycoplasma...</td>\n",
       "      <td>[clinical features culture proven mycoplasma p...</td>\n",
       "      <td>OBJECTIVE: This retrospective chart review des...</td>\n",
       "      <td>[objective retrospective chart review epidemio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n",
       "      <td>[nitric oxide pro inflammatory mediator lung d...</td>\n",
       "      <td>Inflammatory diseases of the respiratory tract...</td>\n",
       "      <td>[inflammatory diseases respiratory tract commo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Surfactant protein-D and pulmonary host defense</td>\n",
       "      <td>[surfactant protein pulmonary host defense]</td>\n",
       "      <td>Surfactant protein-D (SP-D) participates in th...</td>\n",
       "      <td>[surfactant protein participates response micr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Role of endothelin-1 in lung disease</td>\n",
       "      <td>[role endothelin lung disease]</td>\n",
       "      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n",
       "      <td>[endothelin amino acid peptide diverse biologi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene expression in epithelial cells in respons...</td>\n",
       "      <td>[gene expression epithelial cells response pne...</td>\n",
       "      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n",
       "      <td>[respiratory syncytial virus rsv pneumonia vir...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>Mannose-binding lectin deficiency and acute ex...</td>\n",
       "      <td>[mannose binding lectin deficiency acute exace...</td>\n",
       "      <td>BACKGROUND: Mannose-binding lectin is a collec...</td>\n",
       "      <td>[background mannose binding lectin collectin h...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>The changing phenotype of microglia from homeo...</td>\n",
       "      <td>[changing phenotype microglia homeostasis dise...</td>\n",
       "      <td>It has been nearly a century since the early d...</td>\n",
       "      <td>[century early description microglia rio many ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>Diversity of Salmonella spp. serovars isolated...</td>\n",
       "      <td>[diversity salmonella spp, serovars isolated i...</td>\n",
       "      <td>BACKGROUND: Salmonellosis in water buffalo (Bu...</td>\n",
       "      <td>[, , present study, water buffalo calves typhi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>Severe Childhood Malaria Syndromes Defined by ...</td>\n",
       "      <td>[severe childhood malaria syndromes defined pl...</td>\n",
       "      <td>BACKGROUND: Cerebral malaria (CM) and severe m...</td>\n",
       "      <td>[background cerebral malaria serious life clin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>Predicting pseudoknotted structures across two...</td>\n",
       "      <td>[predicting pseudoknotted structures across tw...</td>\n",
       "      <td>Motivation: Laboratory RNA structure determina...</td>\n",
       "      <td>[motivation laboratory rna structure determina...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 title  \\\n",
       "0    Clinical features of culture-proven Mycoplasma...   \n",
       "1    Nitric oxide: a pro-inflammatory mediator in l...   \n",
       "2      Surfactant protein-D and pulmonary host defense   \n",
       "3                 Role of endothelin-1 in lung disease   \n",
       "4    Gene expression in epithelial cells in respons...   \n",
       "..                                                 ...   \n",
       "995  Mannose-binding lectin deficiency and acute ex...   \n",
       "996  The changing phenotype of microglia from homeo...   \n",
       "997  Diversity of Salmonella spp. serovars isolated...   \n",
       "998  Severe Childhood Malaria Syndromes Defined by ...   \n",
       "999  Predicting pseudoknotted structures across two...   \n",
       "\n",
       "                                              pr_title  \\\n",
       "0    [clinical features culture proven mycoplasma p...   \n",
       "1    [nitric oxide pro inflammatory mediator lung d...   \n",
       "2          [surfactant protein pulmonary host defense]   \n",
       "3                       [role endothelin lung disease]   \n",
       "4    [gene expression epithelial cells response pne...   \n",
       "..                                                 ...   \n",
       "995  [mannose binding lectin deficiency acute exace...   \n",
       "996  [changing phenotype microglia homeostasis dise...   \n",
       "997  [diversity salmonella spp, serovars isolated i...   \n",
       "998  [severe childhood malaria syndromes defined pl...   \n",
       "999  [predicting pseudoknotted structures across tw...   \n",
       "\n",
       "                                              abstract  \\\n",
       "0    OBJECTIVE: This retrospective chart review des...   \n",
       "1    Inflammatory diseases of the respiratory tract...   \n",
       "2    Surfactant protein-D (SP-D) participates in th...   \n",
       "3    Endothelin-1 (ET-1) is a 21 amino acid peptide...   \n",
       "4    Respiratory syncytial virus (RSV) and pneumoni...   \n",
       "..                                                 ...   \n",
       "995  BACKGROUND: Mannose-binding lectin is a collec...   \n",
       "996  It has been nearly a century since the early d...   \n",
       "997  BACKGROUND: Salmonellosis in water buffalo (Bu...   \n",
       "998  BACKGROUND: Cerebral malaria (CM) and severe m...   \n",
       "999  Motivation: Laboratory RNA structure determina...   \n",
       "\n",
       "                                           pr_abstract  \n",
       "0    [objective retrospective chart review epidemio...  \n",
       "1    [inflammatory diseases respiratory tract commo...  \n",
       "2    [surfactant protein participates response micr...  \n",
       "3    [endothelin amino acid peptide diverse biologi...  \n",
       "4    [respiratory syncytial virus rsv pneumonia vir...  \n",
       "..                                                 ...  \n",
       "995  [background mannose binding lectin collectin h...  \n",
       "996  [century early description microglia rio many ...  \n",
       "997  [, , present study, water buffalo calves typhi...  \n",
       "998  [background cerebral malaria serious life clin...  \n",
       "999  [motivation laboratory rna structure determina...  \n",
       "\n",
       "[1000 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Text lemmatisation\n",
      "Comparison of Text Data after Applied Lemmatisation\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>pr_title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>pr_abstract</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Clinical features of culture-proven Mycoplasma...</td>\n",
       "      <td>[clinical feature culture proven mycoplasma pn...</td>\n",
       "      <td>OBJECTIVE: This retrospective chart review des...</td>\n",
       "      <td>[objective retrospective chart review epidemio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Nitric oxide: a pro-inflammatory mediator in l...</td>\n",
       "      <td>[nitric oxide pro inflammatory mediator lung d...</td>\n",
       "      <td>Inflammatory diseases of the respiratory tract...</td>\n",
       "      <td>[inflammatory disease respiratory tract common...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Surfactant protein-D and pulmonary host defense</td>\n",
       "      <td>[surfactant protein pulmonary host defense]</td>\n",
       "      <td>Surfactant protein-D (SP-D) participates in th...</td>\n",
       "      <td>[surfactant protein participates response micr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Role of endothelin-1 in lung disease</td>\n",
       "      <td>[role endothelin lung disease]</td>\n",
       "      <td>Endothelin-1 (ET-1) is a 21 amino acid peptide...</td>\n",
       "      <td>[endothelin amino acid peptide diverse biologi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gene expression in epithelial cells in respons...</td>\n",
       "      <td>[gene expression epithelial cell response pneu...</td>\n",
       "      <td>Respiratory syncytial virus (RSV) and pneumoni...</td>\n",
       "      <td>[respiratory syncytial virus rsv pneumonia vir...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>Mannose-binding lectin deficiency and acute ex...</td>\n",
       "      <td>[mannose binding lectin deficiency acute exace...</td>\n",
       "      <td>BACKGROUND: Mannose-binding lectin is a collec...</td>\n",
       "      <td>[background mannose binding lectin collectin h...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>The changing phenotype of microglia from homeo...</td>\n",
       "      <td>[changing phenotype microglia homeostasis dise...</td>\n",
       "      <td>It has been nearly a century since the early d...</td>\n",
       "      <td>[century early description microglia rio many ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>Diversity of Salmonella spp. serovars isolated...</td>\n",
       "      <td>[diversity salmonella spp, serovars isolated i...</td>\n",
       "      <td>BACKGROUND: Salmonellosis in water buffalo (Bu...</td>\n",
       "      <td>[, , present study, water buffalo calf typhimu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>Severe Childhood Malaria Syndromes Defined by ...</td>\n",
       "      <td>[severe childhood malaria syndrome defined pla...</td>\n",
       "      <td>BACKGROUND: Cerebral malaria (CM) and severe m...</td>\n",
       "      <td>[background cerebral malaria serious life clin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>Predicting pseudoknotted structures across two...</td>\n",
       "      <td>[predicting pseudoknotted structure across two...</td>\n",
       "      <td>Motivation: Laboratory RNA structure determina...</td>\n",
       "      <td>[motivation laboratory rna structure determina...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 title  \\\n",
       "0    Clinical features of culture-proven Mycoplasma...   \n",
       "1    Nitric oxide: a pro-inflammatory mediator in l...   \n",
       "2      Surfactant protein-D and pulmonary host defense   \n",
       "3                 Role of endothelin-1 in lung disease   \n",
       "4    Gene expression in epithelial cells in respons...   \n",
       "..                                                 ...   \n",
       "995  Mannose-binding lectin deficiency and acute ex...   \n",
       "996  The changing phenotype of microglia from homeo...   \n",
       "997  Diversity of Salmonella spp. serovars isolated...   \n",
       "998  Severe Childhood Malaria Syndromes Defined by ...   \n",
       "999  Predicting pseudoknotted structures across two...   \n",
       "\n",
       "                                              pr_title  \\\n",
       "0    [clinical feature culture proven mycoplasma pn...   \n",
       "1    [nitric oxide pro inflammatory mediator lung d...   \n",
       "2          [surfactant protein pulmonary host defense]   \n",
       "3                       [role endothelin lung disease]   \n",
       "4    [gene expression epithelial cell response pneu...   \n",
       "..                                                 ...   \n",
       "995  [mannose binding lectin deficiency acute exace...   \n",
       "996  [changing phenotype microglia homeostasis dise...   \n",
       "997  [diversity salmonella spp, serovars isolated i...   \n",
       "998  [severe childhood malaria syndrome defined pla...   \n",
       "999  [predicting pseudoknotted structure across two...   \n",
       "\n",
       "                                              abstract  \\\n",
       "0    OBJECTIVE: This retrospective chart review des...   \n",
       "1    Inflammatory diseases of the respiratory tract...   \n",
       "2    Surfactant protein-D (SP-D) participates in th...   \n",
       "3    Endothelin-1 (ET-1) is a 21 amino acid peptide...   \n",
       "4    Respiratory syncytial virus (RSV) and pneumoni...   \n",
       "..                                                 ...   \n",
       "995  BACKGROUND: Mannose-binding lectin is a collec...   \n",
       "996  It has been nearly a century since the early d...   \n",
       "997  BACKGROUND: Salmonellosis in water buffalo (Bu...   \n",
       "998  BACKGROUND: Cerebral malaria (CM) and severe m...   \n",
       "999  Motivation: Laboratory RNA structure determina...   \n",
       "\n",
       "                                           pr_abstract  \n",
       "0    [objective retrospective chart review epidemio...  \n",
       "1    [inflammatory disease respiratory tract common...  \n",
       "2    [surfactant protein participates response micr...  \n",
       "3    [endothelin amino acid peptide diverse biologi...  \n",
       "4    [respiratory syncytial virus rsv pneumonia vir...  \n",
       "..                                                 ...  \n",
       "995  [background mannose binding lectin collectin h...  \n",
       "996  [century early description microglia rio many ...  \n",
       "997  [, , present study, water buffalo calf typhimu...  \n",
       "998  [background cerebral malaria serious life clin...  \n",
       "999  [motivation laboratory rna structure determina...  \n",
       "\n",
       "[1000 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating index - covid\n",
      "b'{\\n  \"name\" : \"DESKTOP-K0D65LT\",\\n  \"cluster_name\" : \"elasticsearch\",\\n  \"cluster_uuid\" : \"P8VwA_sWSl-ykHakwwPzXQ\",\\n  \"version\" : {\\n    \"number\" : \"7.10.2\",\\n    \"build_flavor\" : \"default\",\\n    \"build_type\" : \"zip\",\\n    \"build_hash\" : \"747e1cc71def077253878a59143c1f785afa92b9\",\\n    \"build_date\" : \"2021-01-13T00:42:12.435326Z\",\\n    \"build_snapshot\" : false,\\n    \"lucene_version\" : \"8.7.0\",\\n    \"minimum_wire_compatibility_version\" : \"6.8.0\",\\n    \"minimum_index_compatibility_version\" : \"6.0.0-beta1\"\\n  },\\n  \"tagline\" : \"You Know, for Search\"\\n}\\n'\n",
      "Indexing\n",
      "Searching in ElasticSearch\n",
      "Retrieved documents:\n",
      "[{'abstract': 'Inflammatory diseases of the respiratory tract are commonly '\n",
      "              'associated with elevated production of nitric oxide (NO•) and '\n",
      "              'increased indices of NO• -dependent oxidative stress. Although '\n",
      "              'NO• is known to have anti-microbial, anti-inflammatory and '\n",
      "              'anti-oxidant properties, various lines of evidence support the '\n",
      "              'contribution of NO• to lung injury in several disease models. '\n",
      "              'On the basis of biochemical evidence, it is often presumed that '\n",
      "              'such NO• -dependent oxidations are due to the formation of the '\n",
      "              'oxidant peroxynitrite, although alternative mechanisms '\n",
      "              'involving the phagocyte-derived heme proteins myeloperoxidase '\n",
      "              'and eosinophil peroxidase might be operative during conditions '\n",
      "              'of inflammation. Because of the overwhelming literature on NO• '\n",
      "              'generation and activities in the respiratory tract, it would be '\n",
      "              'beyond the scope of this commentary to review this area '\n",
      "              'comprehensively. Instead, it focuses on recent evidence and '\n",
      "              'concepts of the presumed contribution of NO• to inflammatory '\n",
      "              'diseases of the lung.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Vliet, Albert van der; Eiserich, Jason P; Cross, Carroll E',\n",
      "  'cord_uid': '02tnwd4m',\n",
      "  'doi': '10.1186/rr14',\n",
      "  'journal': 'Respir Res',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/6b0567729c2143a66d737eb0a2f63f2dce2e5a7d.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC59543.xml.json',\n",
      "  'pmcid': 'PMC59543',\n",
      "  'pr_abstract': ['inflammatory disease respiratory tract commonly associated '\n",
      "                  'elevated production nitric oxide index dependent oxidative '\n",
      "                  'stress',\n",
      "                  'property various line evidence support contribution lung '\n",
      "                  'injury several disease model',\n",
      "                  'basis biochemical evidence dependent oxidation formation '\n",
      "                  'alternative mechanism phagocyte heme protein eosinophil '\n",
      "                  'peroxidase condition inflammation',\n",
      "                  'overwhelming literature generation activity respiratory '\n",
      "                  'tract scope commentary review area',\n",
      "                  'recent evidence concept presumed contribution inflammatory '\n",
      "                  'disease lung',\n",
      "                  'literature generation respiratory tract area',\n",
      "                  'basis evidence dependent formation although alternative '\n",
      "                  'condition inflammation',\n",
      "                  'inflammatory disease respiratory tract commonly nitric '\n",
      "                  'oxide dependent oxidative stress',\n",
      "                  'although anti anti inflammatory anti property line evidence '\n",
      "                  'lung injury',\n",
      "                  'evidence inflammatory disease lung'],\n",
      "  'pr_title': ['nitric oxide pro inflammatory mediator lung disease'],\n",
      "  'publish_time': '2000-08-15',\n",
      "  'pubmed_id': '11667967',\n",
      "  's2_id': None,\n",
      "  'sha': '6b0567729c2143a66d737eb0a2f63f2dce2e5a7d',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Nitric oxide: a pro-inflammatory mediator in lung disease?',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC59543/',\n",
      "  'who_covidence_id': None}]\n",
      "Retrieved documents:\n",
      "[{'abstract': 'Inflammatory diseases of the respiratory tract are commonly '\n",
      "              'associated with elevated production of nitric oxide (NO•) and '\n",
      "              'increased indices of NO• -dependent oxidative stress. Although '\n",
      "              'NO• is known to have anti-microbial, anti-inflammatory and '\n",
      "              'anti-oxidant properties, various lines of evidence support the '\n",
      "              'contribution of NO• to lung injury in several disease models. '\n",
      "              'On the basis of biochemical evidence, it is often presumed that '\n",
      "              'such NO• -dependent oxidations are due to the formation of the '\n",
      "              'oxidant peroxynitrite, although alternative mechanisms '\n",
      "              'involving the phagocyte-derived heme proteins myeloperoxidase '\n",
      "              'and eosinophil peroxidase might be operative during conditions '\n",
      "              'of inflammation. Because of the overwhelming literature on NO• '\n",
      "              'generation and activities in the respiratory tract, it would be '\n",
      "              'beyond the scope of this commentary to review this area '\n",
      "              'comprehensively. Instead, it focuses on recent evidence and '\n",
      "              'concepts of the presumed contribution of NO• to inflammatory '\n",
      "              'diseases of the lung.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Vliet, Albert van der; Eiserich, Jason P; Cross, Carroll E',\n",
      "  'cord_uid': '02tnwd4m',\n",
      "  'doi': '10.1186/rr14',\n",
      "  'journal': 'Respir Res',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/6b0567729c2143a66d737eb0a2f63f2dce2e5a7d.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC59543.xml.json',\n",
      "  'pmcid': 'PMC59543',\n",
      "  'pr_abstract': ['inflammatory disease respiratory tract commonly associated '\n",
      "                  'elevated production nitric oxide index dependent oxidative '\n",
      "                  'stress',\n",
      "                  'property various line evidence support contribution lung '\n",
      "                  'injury several disease model',\n",
      "                  'basis biochemical evidence dependent oxidation formation '\n",
      "                  'alternative mechanism phagocyte heme protein eosinophil '\n",
      "                  'peroxidase condition inflammation',\n",
      "                  'overwhelming literature generation activity respiratory '\n",
      "                  'tract scope commentary review area',\n",
      "                  'recent evidence concept presumed contribution inflammatory '\n",
      "                  'disease lung',\n",
      "                  'literature generation respiratory tract area',\n",
      "                  'basis evidence dependent formation although alternative '\n",
      "                  'condition inflammation',\n",
      "                  'inflammatory disease respiratory tract commonly nitric '\n",
      "                  'oxide dependent oxidative stress',\n",
      "                  'although anti anti inflammatory anti property line evidence '\n",
      "                  'lung injury',\n",
      "                  'evidence inflammatory disease lung'],\n",
      "  'pr_title': ['nitric oxide pro inflammatory mediator lung disease'],\n",
      "  'publish_time': '2000-08-15',\n",
      "  'pubmed_id': '11667967',\n",
      "  's2_id': None,\n",
      "  'sha': '6b0567729c2143a66d737eb0a2f63f2dce2e5a7d',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Nitric oxide: a pro-inflammatory mediator in lung disease?',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC59543/',\n",
      "  'who_covidence_id': None}]\n"
     ]
    }
   ],
   "source": [
    "documents,es=run_program(path = 'metadata.csv',es_index=\"covid\", query={\n",
    "              \"query\": {\n",
    "                \"match_phrase\":{\"publish_time\":\"2000-08-15\"}\n",
    "              }\n",
    "            })\n",
    "print(\"Retrieved documents:\")\n",
    "pprint.pprint(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieved documents:\n",
      "[{'abstract': 'Inflammatory diseases of the respiratory tract are commonly '\n",
      "              'associated with elevated production of nitric oxide (NO•) and '\n",
      "              'increased indices of NO• -dependent oxidative stress. Although '\n",
      "              'NO• is known to have anti-microbial, anti-inflammatory and '\n",
      "              'anti-oxidant properties, various lines of evidence support the '\n",
      "              'contribution of NO• to lung injury in several disease models. '\n",
      "              'On the basis of biochemical evidence, it is often presumed that '\n",
      "              'such NO• -dependent oxidations are due to the formation of the '\n",
      "              'oxidant peroxynitrite, although alternative mechanisms '\n",
      "              'involving the phagocyte-derived heme proteins myeloperoxidase '\n",
      "              'and eosinophil peroxidase might be operative during conditions '\n",
      "              'of inflammation. Because of the overwhelming literature on NO• '\n",
      "              'generation and activities in the respiratory tract, it would be '\n",
      "              'beyond the scope of this commentary to review this area '\n",
      "              'comprehensively. Instead, it focuses on recent evidence and '\n",
      "              'concepts of the presumed contribution of NO• to inflammatory '\n",
      "              'diseases of the lung.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Vliet, Albert van der; Eiserich, Jason P; Cross, Carroll E',\n",
      "  'cord_uid': '02tnwd4m',\n",
      "  'doi': '10.1186/rr14',\n",
      "  'journal': 'Respir Res',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/6b0567729c2143a66d737eb0a2f63f2dce2e5a7d.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC59543.xml.json',\n",
      "  'pmcid': 'PMC59543',\n",
      "  'pr_abstract': ['inflammatory disease respiratory tract commonly associated '\n",
      "                  'elevated production nitric oxide index dependent oxidative '\n",
      "                  'stress',\n",
      "                  'property various line evidence support contribution lung '\n",
      "                  'injury several disease model',\n",
      "                  'basis biochemical evidence dependent oxidation formation '\n",
      "                  'alternative mechanism phagocyte heme protein eosinophil '\n",
      "                  'peroxidase condition inflammation',\n",
      "                  'overwhelming literature generation activity respiratory '\n",
      "                  'tract scope commentary review area',\n",
      "                  'recent evidence concept presumed contribution inflammatory '\n",
      "                  'disease lung',\n",
      "                  'literature generation respiratory tract area',\n",
      "                  'basis evidence dependent formation although alternative '\n",
      "                  'condition inflammation',\n",
      "                  'inflammatory disease respiratory tract commonly nitric '\n",
      "                  'oxide dependent oxidative stress',\n",
      "                  'although anti anti inflammatory anti property line evidence '\n",
      "                  'lung injury',\n",
      "                  'evidence inflammatory disease lung'],\n",
      "  'pr_title': ['nitric oxide pro inflammatory mediator lung disease'],\n",
      "  'publish_time': '2000-08-15',\n",
      "  'pubmed_id': '11667967',\n",
      "  's2_id': None,\n",
      "  'sha': '6b0567729c2143a66d737eb0a2f63f2dce2e5a7d',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Nitric oxide: a pro-inflammatory mediator in lung disease?',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC59543/',\n",
      "  'who_covidence_id': None}]\n"
     ]
    }
   ],
   "source": [
    "documents=search(es, es_index=\"covid\",\n",
    "    query={\n",
    "          \"query\": {\n",
    "            \"match_phrase\":{\"publish_time\":\"2000-08-15\"}\n",
    "          }\n",
    "        })\n",
    "print(\"Retrieved documents:\")\n",
    "pprint.pprint(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieved documents:\n",
      "[{'abstract': 'Sapovirus is a genus of caliciviruses that are known to cause '\n",
      "              'enteric disease in humans and animals. There is considerable '\n",
      "              'genetic diversity among the sapoviruses, which are classified '\n",
      "              'into different genogroups based on phylogenetic analysis of the '\n",
      "              'full-length capsid protein sequence. While several mammalian '\n",
      "              'species, including humans, pigs, minks, and dogs, have been '\n",
      "              'identified as animal hosts for sapoviruses, there were no '\n",
      "              'reports of sapoviruses in bats in spite of their biological '\n",
      "              'diversity. In this report, we present the results of a targeted '\n",
      "              'surveillance study in different bat species in Hong Kong. Five '\n",
      "              'of the 321 specimens from the bat species, Hipposideros pomona, '\n",
      "              'were found to be positive for sapoviruses by RT-PCR. Complete '\n",
      "              'or nearly full-length genome sequences of approximately 7.7 kb '\n",
      "              'in length were obtained for three strains, which showed similar '\n",
      "              'organization of the genome compared to other sapoviruses. '\n",
      "              'Interestingly, they possess many genomic features atypical of '\n",
      "              'most sapoviruses, like high G+C content and minimal CpG '\n",
      "              'suppression. Phylogenetic analysis of the viral proteins '\n",
      "              'suggested that the bat sapovirus descended from an ancestral '\n",
      "              'sapovirus lineage and is most closely related to the porcine '\n",
      "              'sapoviruses. Codon usage analysis showed that the bat sapovirus '\n",
      "              'genome has greater codon usage bias relative to other sapovirus '\n",
      "              'genomes. In summary, we report the discovery and genomic '\n",
      "              'characterization of the first bat calicivirus, which appears to '\n",
      "              'have evolved under different conditions after early divergence '\n",
      "              'from other sapovirus lineages.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Tse, Herman; Chan, Wan-Mui; Li, Kenneth S. M.; Lau, Susanna K. '\n",
      "             'P.; Woo, Patrick C. Y.; Yuen, Kwok-Yung',\n",
      "  'cord_uid': 'alyn1i00',\n",
      "  'doi': '10.1371/journal.pone.0034987',\n",
      "  'journal': 'PLoS One',\n",
      "  'license': 'cc-by',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/dbf7ea2d45fff41a73853c26e5fca9862fd0edcc.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC3325917.xml.json',\n",
      "  'pmcid': 'PMC3325917',\n",
      "  'pr_abstract': ['sapovirus genus caliciviruses cause enteric disease human '\n",
      "                  'animal',\n",
      "                  'considerable genetic diversity sapoviruses classified '\n",
      "                  'different genogroups based phylogenetic analysis full '\n",
      "                  'length capsid protein sequence',\n",
      "                  'several mammalian specie human mink dog animal host '\n",
      "                  'sapoviruses report sapoviruses bat biological diversity',\n",
      "                  'surveillance study specie hong kong',\n",
      "                  'specie positive sapoviruses',\n",
      "                  'full length genome sequence strain similar organization '\n",
      "                  'genome sapoviruses',\n",
      "                  'many genomic feature atypical sapoviruses',\n",
      "                  'phylogenetic analysis viral protein bat sapovirus ancestral '\n",
      "                  'sapovirus lineage closely related porcine sapoviruses',\n",
      "                  'codon usage analysis bat sapovirus genome greater codon '\n",
      "                  'usage bias relative sapovirus genome',\n",
      "                  'summary report discovery genomic characterization first bat '\n",
      "                  'calicivirus different condition early divergence sapovirus '\n",
      "                  'lineage',\n",
      "                  'specie human report',\n",
      "                  'full length genome sequence strain genome',\n",
      "                  'codon usage genome codon usage genome',\n",
      "                  'different phylogenetic analysis full length',\n",
      "                  'cause human animal',\n",
      "                  'report different specie hong kong',\n",
      "                  'phylogenetic analysis viral protein',\n",
      "                  'genomic',\n",
      "                  'report genomic different',\n",
      "                  'specie'],\n",
      "  'pr_title': ['discovery genomic characterization novel bat sapovirus unusual '\n",
      "               'genomic feature phylogenetic position'],\n",
      "  'publish_time': '2012-04-13',\n",
      "  'pubmed_id': '22514697',\n",
      "  's2_id': None,\n",
      "  'sha': 'dbf7ea2d45fff41a73853c26e5fca9862fd0edcc',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Discovery and Genomic Characterization of a Novel Bat Sapovirus '\n",
      "           'with Unusual Genomic Features and Phylogenetic Position',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3325917/',\n",
      "  'who_covidence_id': None}]\n"
     ]
    }
   ],
   "source": [
    "documents=search(es, es_index=\"covid\",\n",
    "    query={\n",
    "          \"query\": {\n",
    "            \"match_phrase\":{\"pr_abstract\":\"biological diversity\"}\n",
    "          }\n",
    "        })\n",
    "print(\"Retrieved documents:\")\n",
    "pprint.pprint(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieved documents:\n",
      "[{'abstract': 'Horizontal DNA transfer is an important factor of evolution and '\n",
      "              'participates in biological diversity. Unfortunately, the '\n",
      "              'location and length of horizontal transfers (HTs) are known for '\n",
      "              'very few species. The usage of short oligonucleotides in a '\n",
      "              'sequence (the so-called genomic signature) has been shown to be '\n",
      "              'species-specific even in DNA fragments as short as 1 kb. The '\n",
      "              'genomic signature is therefore proposed as a tool to detect '\n",
      "              'HTs. Since DNA transfers originate from species with a '\n",
      "              'signature different from those of the recipient species, the '\n",
      "              'analysis of local variations of signature along recipient '\n",
      "              'genome may allow for detecting exogenous DNA. The strategy '\n",
      "              'consists in (i) scanning the genome with a sliding window, and '\n",
      "              'calculating the corresponding local signature (ii) evaluating '\n",
      "              'its deviation from the signature of the whole genome and (iii) '\n",
      "              'looking for similar signatures in a database of genomic '\n",
      "              'signatures. A total of 22 prokaryote genomes are analyzed in '\n",
      "              'this way. It has been observed that atypical regions make up '\n",
      "              '∼6% of each genome on the average. Most of the claimed HTs as '\n",
      "              'well as new ones are detected. The origin of putative DNA '\n",
      "              'transfers is looked for among ∼12 000 species. Donor species '\n",
      "              'are proposed and sometimes strongly suggested, considering '\n",
      "              'similarity of signatures. Among the species studied, Bacillus '\n",
      "              'subtilis, Haemophilus Influenzae and Escherichia coli are '\n",
      "              'investigated by many authors and give the opportunity to '\n",
      "              'perform a thorough comparison of most of the bioinformatics '\n",
      "              'methods used to detect HTs.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Dufraigne, Christine; Fertil, Bernard; Lespinats, Sylvain; '\n",
      "             'Giron, Alain; Deschavanne, Patrick',\n",
      "  'cord_uid': 'yba7mdtb',\n",
      "  'doi': '10.1093/nar/gni004',\n",
      "  'journal': 'Nucleic Acids Res',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/f1d1b9694aa43c837d9b758cb2d45d8a24d293e3.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC546175.xml.json',\n",
      "  'pmcid': 'PMC546175',\n",
      "  'pr_abstract': ['dna important factor evolution',\n",
      "                  'hts specie',\n",
      "                  'short oligonucleotides sequence genomic signature specie',\n",
      "                  'genomic signature tool hts',\n",
      "                  'dna transfer specie different recipient specie analysis '\n",
      "                  'local variation signature recipient genome dna',\n",
      "                  'strategy genome sliding window deviation signature whole '\n",
      "                  'genome database',\n",
      "                  'genome analyzed way',\n",
      "                  'observed atypical region genome average',\n",
      "                  'hts new one',\n",
      "                  'origin putative dna transfer specie',\n",
      "                  'specie similarity signature',\n",
      "                  'specie studied bacillus subtilis haemophilus many author '\n",
      "                  'thorough comparison bioinformatics method hts',\n",
      "                  'genome',\n",
      "                  'specie proposed',\n",
      "                  'dna',\n",
      "                  'genome',\n",
      "                  'short genomic dna short',\n",
      "                  'specie',\n",
      "                  'dna specie',\n",
      "                  'genomic proposed tool detect',\n",
      "                  'dna specie specie local genome dna',\n",
      "                  'specie detect',\n",
      "                  'strategy genome local whole genome database genomic',\n",
      "                  ''],\n",
      "  'pr_title': ['detection characterization horizontal transfer prokaryote '\n",
      "               'using genomic signature'],\n",
      "  'publish_time': '2005-01-13',\n",
      "  'pubmed_id': '15653627',\n",
      "  's2_id': None,\n",
      "  'sha': 'f1d1b9694aa43c837d9b758cb2d45d8a24d293e3',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Detection and characterization of horizontal transfers in '\n",
      "           'prokaryotes using genomic signature',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC546175/',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'Sapovirus is a genus of caliciviruses that are known to cause '\n",
      "              'enteric disease in humans and animals. There is considerable '\n",
      "              'genetic diversity among the sapoviruses, which are classified '\n",
      "              'into different genogroups based on phylogenetic analysis of the '\n",
      "              'full-length capsid protein sequence. While several mammalian '\n",
      "              'species, including humans, pigs, minks, and dogs, have been '\n",
      "              'identified as animal hosts for sapoviruses, there were no '\n",
      "              'reports of sapoviruses in bats in spite of their biological '\n",
      "              'diversity. In this report, we present the results of a targeted '\n",
      "              'surveillance study in different bat species in Hong Kong. Five '\n",
      "              'of the 321 specimens from the bat species, Hipposideros pomona, '\n",
      "              'were found to be positive for sapoviruses by RT-PCR. Complete '\n",
      "              'or nearly full-length genome sequences of approximately 7.7 kb '\n",
      "              'in length were obtained for three strains, which showed similar '\n",
      "              'organization of the genome compared to other sapoviruses. '\n",
      "              'Interestingly, they possess many genomic features atypical of '\n",
      "              'most sapoviruses, like high G+C content and minimal CpG '\n",
      "              'suppression. Phylogenetic analysis of the viral proteins '\n",
      "              'suggested that the bat sapovirus descended from an ancestral '\n",
      "              'sapovirus lineage and is most closely related to the porcine '\n",
      "              'sapoviruses. Codon usage analysis showed that the bat sapovirus '\n",
      "              'genome has greater codon usage bias relative to other sapovirus '\n",
      "              'genomes. In summary, we report the discovery and genomic '\n",
      "              'characterization of the first bat calicivirus, which appears to '\n",
      "              'have evolved under different conditions after early divergence '\n",
      "              'from other sapovirus lineages.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Tse, Herman; Chan, Wan-Mui; Li, Kenneth S. M.; Lau, Susanna K. '\n",
      "             'P.; Woo, Patrick C. Y.; Yuen, Kwok-Yung',\n",
      "  'cord_uid': 'alyn1i00',\n",
      "  'doi': '10.1371/journal.pone.0034987',\n",
      "  'journal': 'PLoS One',\n",
      "  'license': 'cc-by',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/dbf7ea2d45fff41a73853c26e5fca9862fd0edcc.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC3325917.xml.json',\n",
      "  'pmcid': 'PMC3325917',\n",
      "  'pr_abstract': ['sapovirus genus caliciviruses cause enteric disease human '\n",
      "                  'animal',\n",
      "                  'considerable genetic diversity sapoviruses classified '\n",
      "                  'different genogroups based phylogenetic analysis full '\n",
      "                  'length capsid protein sequence',\n",
      "                  'several mammalian specie human mink dog animal host '\n",
      "                  'sapoviruses report sapoviruses bat biological diversity',\n",
      "                  'surveillance study specie hong kong',\n",
      "                  'specie positive sapoviruses',\n",
      "                  'full length genome sequence strain similar organization '\n",
      "                  'genome sapoviruses',\n",
      "                  'many genomic feature atypical sapoviruses',\n",
      "                  'phylogenetic analysis viral protein bat sapovirus ancestral '\n",
      "                  'sapovirus lineage closely related porcine sapoviruses',\n",
      "                  'codon usage analysis bat sapovirus genome greater codon '\n",
      "                  'usage bias relative sapovirus genome',\n",
      "                  'summary report discovery genomic characterization first bat '\n",
      "                  'calicivirus different condition early divergence sapovirus '\n",
      "                  'lineage',\n",
      "                  'specie human report',\n",
      "                  'full length genome sequence strain genome',\n",
      "                  'codon usage genome codon usage genome',\n",
      "                  'different phylogenetic analysis full length',\n",
      "                  'cause human animal',\n",
      "                  'report different specie hong kong',\n",
      "                  'phylogenetic analysis viral protein',\n",
      "                  'genomic',\n",
      "                  'report genomic different',\n",
      "                  'specie'],\n",
      "  'pr_title': ['discovery genomic characterization novel bat sapovirus unusual '\n",
      "               'genomic feature phylogenetic position'],\n",
      "  'publish_time': '2012-04-13',\n",
      "  'pubmed_id': '22514697',\n",
      "  's2_id': None,\n",
      "  'sha': 'dbf7ea2d45fff41a73853c26e5fca9862fd0edcc',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Discovery and Genomic Characterization of a Novel Bat Sapovirus '\n",
      "           'with Unusual Genomic Features and Phylogenetic Position',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3325917/',\n",
      "  'who_covidence_id': None}]\n"
     ]
    }
   ],
   "source": [
    "documents=search(es, es_index=\"covid\",\n",
    "    query={\n",
    "          \"query\": {\n",
    "            \"match_phrase\":{\"abstract\":\"biological diversity\"}\n",
    "          }\n",
    "        })\n",
    "print(\"Retrieved documents:\")\n",
    "pprint.pprint(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Retrieved documents:\n",
      "[{'abstract': 'Nidovirus subgenomic mRNAs contain a leader sequence derived '\n",
      "              'from the 5′ end of the genome fused to different sequences '\n",
      "              '(‘bodies’) derived from the 3′ end. Their generation involves a '\n",
      "              'unique mechanism of discontinuous subgenomic RNA synthesis that '\n",
      "              'resembles copy-choice RNA recombination. During this process, '\n",
      "              'the nascent RNA strand is transferred from one site in the '\n",
      "              'template to another, during either plus or minus strand '\n",
      "              'synthesis, to yield subgenomic RNA molecules. Central to this '\n",
      "              'process are transcription-regulating sequences (TRSs), which '\n",
      "              'are present at both template sites and ensure the fidelity of '\n",
      "              'strand transfer. Here we present results of a comprehensive '\n",
      "              'co-variation mutagenesis study of equine arteritis virus TRSs, '\n",
      "              'demonstrating that discontinuous RNA synthesis depends not only '\n",
      "              'on base pairing between sense leader TRS and antisense body '\n",
      "              'TRS, but also on the primary sequence of the body TRS. While '\n",
      "              'the leader TRS merely plays a targeting role for strand '\n",
      "              'transfer, the body TRS fulfils multiple functions. The '\n",
      "              'sequences of mRNA leader–body junctions of TRS mutants strongly '\n",
      "              'suggested that the discontinuous step occurs during minus '\n",
      "              'strand synthesis.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Pasternak, Alexander O.; van den Born, Erwin; Spaan, Willy J.M.; '\n",
      "             'Snijder, Eric J.',\n",
      "  'cord_uid': 'zjufx4fo',\n",
      "  'doi': '10.1093/emboj/20.24.7220',\n",
      "  'journal': 'The EMBO Journal',\n",
      "  'license': 'green-oa',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/b2897e1277f56641193a6db73825f707eed3e4c9.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC125340.xml.json',\n",
      "  'pmcid': 'PMC125340',\n",
      "  'pr_abstract': ['nidovirus subgenomic mrna leader sequence end genome '\n",
      "                  'different sequence body end',\n",
      "                  'unique mechanism discontinuous subgenomic rna synthesis rna',\n",
      "                  'process rna site strand synthesis subgenomic rna molecule',\n",
      "                  'central process transcription sequence trss present '\n",
      "                  'template site',\n",
      "                  'present result comprehensive variation mutagenesis study '\n",
      "                  'equine arteritis virus trss discontinuous rna synthesis '\n",
      "                  'base pairing sense leader trs antisense body primary '\n",
      "                  'sequence body',\n",
      "                  'role strand transfer body trs fulfils multiple function',\n",
      "                  'sequence mrna leader body junction trs mutant discontinuous '\n",
      "                  'step strand synthesis',\n",
      "                  'process rna strand site strand synthesis rna',\n",
      "                  'rna synthesis rna recombination',\n",
      "                  'process sequence present strand transfer',\n",
      "                  'sequence derived end sequence derived end',\n",
      "                  'sequence strand synthesis',\n",
      "                  'present rna synthesis sequence',\n",
      "                  'strand transfer'],\n",
      "  'pr_title': ['sequence requirement rna strand transfer nidovirus '\n",
      "               'discontinuous subgenomic rna synthesis'],\n",
      "  'publish_time': '2001-12-17',\n",
      "  'pubmed_id': '11742998',\n",
      "  's2_id': None,\n",
      "  'sha': 'b2897e1277f56641193a6db73825f707eed3e4c9',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Sequence requirements for RNA strand transfer during nidovirus '\n",
      "           'discontinuous subgenomic RNA synthesis',\n",
      "  'url': 'http://europepmc.org/articles/pmc125340?pdf=render',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'Recent evidence suggests that critically ill patients are able '\n",
      "              'to tolerate lower levels of haemoglobin than was previously '\n",
      "              'believed. It is our goal to show that transfusing to a level of '\n",
      "              '100 g/l does not improve mortality and other clinically '\n",
      "              'important outcomes in a critical care setting. Although many '\n",
      "              'questions remain, many laboratory and clinical studies, '\n",
      "              'including a recent randomized controlled trial (RCT), have '\n",
      "              'established that transfusing to normal haemoglobin '\n",
      "              'concentrations does not improve organ failure and mortality in '\n",
      "              'the critically ill patient. In addition, a restrictive '\n",
      "              'transfusion strategy will reduce exposure to allogeneic '\n",
      "              'transfusions, result in more efficient use of red blood cells '\n",
      "              '(RBCs), save blood overall, and decrease health care costs.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Alvarez, Gonzalo; Hébert, Paul C; Szick, Sharyn',\n",
      "  'cord_uid': '5yhe786e',\n",
      "  'doi': '10.1186/cc987',\n",
      "  'journal': 'Crit Care',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/3bb07ea10432f7738413dff9816809cc90f03f99.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC137267.xml.json',\n",
      "  'pmcid': 'PMC137267',\n",
      "  'pr_abstract': ['recent evidence critically ill patient lower level '\n",
      "                  'haemoglobin',\n",
      "                  'goal show transfusing level mortality outcome critical care '\n",
      "                  'setting',\n",
      "                  'many question many laboratory clinical study recent '\n",
      "                  'randomized controlled trial rct transfusing normal '\n",
      "                  'haemoglobin concentration organ failure mortality',\n",
      "                  'addition strategy exposure transfusion blood cell blood '\n",
      "                  'overall decrease health care cost',\n",
      "                  'many remain many clinical study recent randomized '\n",
      "                  'controlled trial established normal concentration improve '\n",
      "                  'organ failure mortality critically ill',\n",
      "                  'recent critically ill able lower',\n",
      "                  'addition strategy reduce exposure blood cell blood health '\n",
      "                  'care',\n",
      "                  'improve mortality outcome critical care'],\n",
      "  'pr_title': ['debate transfusing normal haemoglobin level improve outcome'],\n",
      "  'publish_time': '2001-03-08',\n",
      "  'pubmed_id': '11299062',\n",
      "  's2_id': None,\n",
      "  'sha': '3bb07ea10432f7738413dff9816809cc90f03f99',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Debate: Transfusing to normal haemoglobin levels will not improve '\n",
      "           'outcome',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC137267/',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'The 21st International Symposium on Intensive Care and '\n",
      "              'Emergency Medicine was dominated by the results of recent '\n",
      "              'clinical trials in sepsis and acute respiratory distress '\n",
      "              'syndrome (ARDS). The promise of extracorporeal liver '\n",
      "              'replacement therapy and noninvasive ventilation were other '\n",
      "              'areas of interest. Ethical issues also received attention. '\n",
      "              \"Overall, the 'state of the art' lectures, pro/con debates, \"\n",
      "              'seminars and tutorials were of a high standard. The meeting was '\n",
      "              'marked by a sense of renewed enthusiasm that positive progress '\n",
      "              'is occurring in intensive care medicine.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Ball, Jonathan; Venn, Richard',\n",
      "  'cord_uid': '8zchiykl',\n",
      "  'doi': '10.1186/cc1013',\n",
      "  'journal': 'Crit Care',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/5806726a24dc91de3954001effbdffd7a82d54e2.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC137274.xml.json',\n",
      "  'pmcid': 'PMC137274',\n",
      "  'pr_abstract': ['international symposium intensive care emergency medicine '\n",
      "                  'result recent clinical trial sepsis acute respiratory '\n",
      "                  'distress syndrome ards',\n",
      "                  'extracorporeal liver replacement therapy noninvasive '\n",
      "                  'ventilation area',\n",
      "                  'ethical issue attention',\n",
      "                  'overall state art lecture pro con debate seminar tutorial',\n",
      "                  'meeting sense renewed enthusiasm positive progress '\n",
      "                  'intensive care medicine',\n",
      "                  'overall state standard',\n",
      "                  'international intensive care emergency medicine result '\n",
      "                  'recent clinical trial sepsis acute respiratory distress '\n",
      "                  'syndrome ards',\n",
      "                  'positive intensive care medicine',\n",
      "                  'liver therapy ventilation area interest',\n",
      "                  'received'],\n",
      "  'pr_title': ['international symposium intensive care emergency medicine '\n",
      "               'brussels belgium march'],\n",
      "  'publish_time': '2001-05-02',\n",
      "  'pubmed_id': '11353930',\n",
      "  's2_id': None,\n",
      "  'sha': '5806726a24dc91de3954001effbdffd7a82d54e2',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'The 21st International Symposium on Intensive Care and Emergency '\n",
      "           'Medicine, Brussels, Belgium, 20-23 March 2001',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC137274/',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'Heme oxygenase-1 (HO-1), an inducible stress protein, confers '\n",
      "              'cytoprotection against oxidative stress in vitro and in vivo. '\n",
      "              'In addition to its physiological role in heme degradation, HO-1 '\n",
      "              'may influence a number of cellular processes, including growth, '\n",
      "              'inflammation, and apoptosis. By virtue of anti-inflammatory '\n",
      "              'effects, HO-1 limits tissue damage in response to '\n",
      "              'proinflammatory stimuli and prevents allograft rejection after '\n",
      "              'transplantation. The transcriptional upregulation of HO-1 '\n",
      "              'responds to many agents, such as hypoxia, bacterial '\n",
      "              'lipopolysaccharide, and reactive oxygen/nitrogen species. HO-1 '\n",
      "              'and its constitutively expressed isozyme, heme oxygenase-2, '\n",
      "              'catalyze the rate-limiting step in the conversion of heme to '\n",
      "              'its metabolites, bilirubin IXα, ferrous iron, and carbon '\n",
      "              'monoxide (CO). The mechanisms by which HO-1 provides protection '\n",
      "              'most likely involve its enzymatic reaction products. '\n",
      "              'Remarkably, administration of CO at low concentrations can '\n",
      "              'substitute for HO-1 with respect to anti-inflammatory and '\n",
      "              'anti-apoptotic effects, suggesting a role for CO as a key '\n",
      "              'mediator of HO-1 function. Chronic, low-level, exogenous '\n",
      "              'exposure to CO from cigarette smoking contributes to the '\n",
      "              'importance of CO in pulmonary medicine. The implications of the '\n",
      "              'HO-1/CO system in pulmonary diseases will be discussed in this '\n",
      "              'review, with an emphasis on inflammatory states.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Slebos, Dirk-Jan; Ryter, Stefan W; Choi, Augustine MK',\n",
      "  'cord_uid': '8qnrcgnk',\n",
      "  'doi': '10.1186/1465-9921-4-7',\n",
      "  'journal': 'Respir Res',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/faaf1022ccfe93b032c5608097a53543ba24aedb.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC193681.xml.json',\n",
      "  'pmcid': 'PMC193681',\n",
      "  'pr_abstract': ['heme oxygenase inducible stress protein cytoprotection '\n",
      "                  'oxidative stress',\n",
      "                  'addition role heme degradation number cellular process '\n",
      "                  'growth inflammation apoptosis',\n",
      "                  'anti inflammatory effect tissue damage response '\n",
      "                  'proinflammatory stimulus prevents transplantation',\n",
      "                  'many agent bacterial lipopolysaccharide reactive oxygen '\n",
      "                  'nitrogen specie',\n",
      "                  '',\n",
      "                  'mechanism protection enzymatic reaction product',\n",
      "                  'low concentration anti inflammatory anti apoptotic effect '\n",
      "                  'role function',\n",
      "                  'exposure importance',\n",
      "                  'system disease review emphasis inflammatory state',\n",
      "                  'implication system pulmonary disease state',\n",
      "                  'transcriptional reactive oxygen',\n",
      "                  'mechanism protection reaction',\n",
      "                  'low exposure importance pulmonary medicine',\n",
      "                  'anti inflammatory effect damage',\n",
      "                  'addition role degradation influence growth inflammation '\n",
      "                  'apoptosis',\n",
      "                  'stress oxidative stress vitro vivo',\n",
      "                  'administration low concentration anti inflammatory effect '\n",
      "                  'role function',\n",
      "                  'step'],\n",
      "  'pr_title': ['heme oxygenase carbon monoxide pulmonary medicine'],\n",
      "  'publish_time': '2003-08-07',\n",
      "  'pubmed_id': '12964953',\n",
      "  's2_id': None,\n",
      "  'sha': 'faaf1022ccfe93b032c5608097a53543ba24aedb',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Heme oxygenase-1 and carbon monoxide in pulmonary medicine',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC193681/',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'This report describes the design and implementation of the '\n",
      "              'Real-time Outbreak and Disease Surveillance (RODS) system, a '\n",
      "              'computer-based public health surveillance system for early '\n",
      "              'detection of disease outbreaks. Hospitals send RODS data from '\n",
      "              'clinical encounters over virtual private networks and leased '\n",
      "              'lines using the Health Level 7 (HL7) message protocol. The data '\n",
      "              'are sent in real time. RODS automatically classifies the '\n",
      "              'registration chief complaint from the visit into one of seven '\n",
      "              'syndrome categories using Bayesian classifiers. It stores the '\n",
      "              'data in a relational database, aggregates the data for analysis '\n",
      "              'using data warehousing techniques, applies univariate and '\n",
      "              'multivariate statistical detection algorithms to the data, and '\n",
      "              'alerts users of when the algorithms identify anomalous patterns '\n",
      "              'in the syndrome counts. RODS also has a Web-based user '\n",
      "              'interface that supports temporal and spatial analyses. RODS '\n",
      "              'processes sales of over-the-counter health care products in a '\n",
      "              'similar manner but receives such data in batch mode on a daily '\n",
      "              'basis. RODS was used during the 2002 Winter Olympics and '\n",
      "              'currently operates in two states—Pennsylvania and Utah. It has '\n",
      "              'been and continues to be a resource for implementing, '\n",
      "              'evaluating, and applying new methods of public health '\n",
      "              'surveillance.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Tsui, Fu-Chiang; Espino, Jeremy U.; Dato, Virginia M.; '\n",
      "             'Gesteland, Per H.; Hutman, Judith; Wagner, Michael M.',\n",
      "  'cord_uid': 'jg13scgo',\n",
      "  'doi': '10.1197/jamia.m1345',\n",
      "  'journal': 'Journal of the American Medical Informatics Association',\n",
      "  'license': 'bronze-oa',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/5b44feca5d6ffaaeb66501fa84cc6dd44d06660a.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC212776.xml.json',\n",
      "  'pmcid': 'PMC212776',\n",
      "  'pr_abstract': ['design real time surveillance rod system public health '\n",
      "                  'surveillance system detection',\n",
      "                  'hospital rod data clinical encounter virtual private '\n",
      "                  'network health',\n",
      "                  'data real time',\n",
      "                  'rod registration chief complaint visit',\n",
      "                  'data database data data warehousing technique multivariate '\n",
      "                  'statistical detection algorithm data user anomalous pattern '\n",
      "                  'syndrome',\n",
      "                  'rod also web based user interface analysis',\n",
      "                  'rod sale counter health care product similar manner data '\n",
      "                  'batch mode basis',\n",
      "                  'rod winter olympics state',\n",
      "                  'public health surveillance',\n",
      "                  'design real time system public health detection',\n",
      "                  'data database data data detection data user syndrome',\n",
      "                  'syndrome',\n",
      "                  'analysis',\n",
      "                  'state',\n",
      "                  'data basis',\n",
      "                  'public health surveillance',\n",
      "                  'hospital data health'],\n",
      "  'pr_title': ['technical description rod real time public health surveillance '\n",
      "               'system'],\n",
      "  'publish_time': '2003-09-01',\n",
      "  'pubmed_id': '12807803',\n",
      "  's2_id': None,\n",
      "  'sha': '5b44feca5d6ffaaeb66501fa84cc6dd44d06660a',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Technical Description of RODS: A Real-time Public Health '\n",
      "           'Surveillance System',\n",
      "  'url': 'https://academic.oup.com/jamia/article-pdf/10/5/399/2352016/10-5-399.pdf',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'Regulation of ornithine decarboxylase in vertebrates involves a '\n",
      "              'negative feedback mechanism requiring the protein antizyme. '\n",
      "              'Here we show that a similar mechanism exists in the fission '\n",
      "              'yeast Schizosaccharomyces pombe. The expression of mammalian '\n",
      "              'antizyme genes requires a specific +1 translational frameshift. '\n",
      "              'The efficiency of the frameshift event reflects cellular '\n",
      "              'polyamine levels creating the autoregulatory feedback loop. As '\n",
      "              'shown here, the yeast antizyme gene and several newly '\n",
      "              'identified antizyme genes from different nematodes also require '\n",
      "              'a ribosomal frameshift event for their expression. Twelve '\n",
      "              'nucleotides around the frameshift site are identical between '\n",
      "              'S.pombe and the mammalian counterparts. The core element for '\n",
      "              'this frameshifting is likely to have been present in the last '\n",
      "              'common ancestor of yeast, nematodes and mammals.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Ivanov, Ivaylo P.; Matsufuji, Senya; Murakami, Yasuko; '\n",
      "             'Gesteland, Raymond F.; Atkins, John F.',\n",
      "  'cord_uid': '5tkvsudh',\n",
      "  'doi': '10.1093/emboj/19.8.1907',\n",
      "  'journal': 'EMBO J',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/9d4e3e8eb092d5ed282d0aa4aadcaa8b7165b5e9.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC302018.xml.json',\n",
      "  'pmcid': 'PMC302018',\n",
      "  'pr_abstract': ['regulation ornithine decarboxylase vertebrate negative '\n",
      "                  'feedback mechanism protein antizyme',\n",
      "                  'similar mechanism yeast',\n",
      "                  'expression gene',\n",
      "                  'efficiency cellular polyamine level autoregulatory feedback '\n",
      "                  'loop',\n",
      "                  'yeast antizyme gene several newly identified antizyme gene '\n",
      "                  'different nematode expression',\n",
      "                  'nucleotide frameshift site identical pombe mammalian '\n",
      "                  'counterpart',\n",
      "                  'core element frameshifting last common ancestor yeast '\n",
      "                  'mammal',\n",
      "                  'regulation negative mechanism requiring',\n",
      "                  'mechanism yeast',\n",
      "                  'mammalian',\n",
      "                  'efficiency loop',\n",
      "                  'expression mammalian gene requires',\n",
      "                  'yeast newly identified gene require ribosomal expression',\n",
      "                  'core frameshifting yeast'],\n",
      "  'pr_title': ['conservation polyamine regulation translational frameshifting '\n",
      "               'yeast mammal'],\n",
      "  'publish_time': '2000-04-17',\n",
      "  'pubmed_id': '10775274',\n",
      "  's2_id': None,\n",
      "  'sha': '9d4e3e8eb092d5ed282d0aa4aadcaa8b7165b5e9',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Conservation of polyamine regulation by translational '\n",
      "           'frameshifting from yeast to mammals',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC302018/',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'Heterogeneous nuclear ribonucleoprotein (hnRNP A1) is involved '\n",
      "              'in pre-mRNA splicing in the nucleus and translational '\n",
      "              'regulation in the cytoplasm. In the present study, we '\n",
      "              'demonstrate that hnRNP A1 also participates in the '\n",
      "              'transcription and replication of a cytoplasmic RNA virus, mouse '\n",
      "              'hepatitis virus (MHV). Overexpression of hnRNP A1 accelerated '\n",
      "              'the kinetics of viral RNA synthesis, whereas the expression in '\n",
      "              'the cytoplasm of a dominant-negative hnRNP A1 mutant that lacks '\n",
      "              'the nuclear transport domain significantly delayed it. The '\n",
      "              'hnRNP A1 mutant caused a global inhibition of viral mRNA '\n",
      "              'transcription and genomic replication, and also a preferential '\n",
      "              'inhibition of the replication of defective-interfering RNAs. '\n",
      "              'Similar to the wild-type hnRNP A1, the hnRNP A1 mutant '\n",
      "              'complexed with an MHV polymerase gene product, the nucleocapsid '\n",
      "              'protein and the viral RNA. However, in contrast to the '\n",
      "              'wild-type hnRNP A1, the mutant protein failed to bind a 250 kDa '\n",
      "              'cellular protein, suggesting that the recruitment of cellular '\n",
      "              'proteins by hnRNP A1 is important for MHV RNA synthesis. Our '\n",
      "              'findings establish the importance of cellular factors in viral '\n",
      "              'RNA-dependent RNA synthesis.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Shi, Stephanie T.; Huang, Peiyong; Li, Hsin-Pai; Lai, Michael '\n",
      "             'M.C.',\n",
      "  'cord_uid': '6lvn10f4',\n",
      "  'doi': '10.1093/emboj/19.17.4701',\n",
      "  'journal': 'The EMBO Journal',\n",
      "  'license': 'green-oa',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/14e0cac6e86d62859e6c9f1351ab67466d89e5b3.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC302072.xml.json',\n",
      "  'pmcid': 'PMC302072',\n",
      "  'pr_abstract': ['heterogeneous nuclear ribonucleoprotein hnrnp mrna nucleus '\n",
      "                  'translational regulation',\n",
      "                  'present study demonstrate hnrnp transcription replication '\n",
      "                  'cytoplasmic rna virus mouse hepatitis virus',\n",
      "                  'overexpression hnrnp kinetics viral rna synthesis '\n",
      "                  'expression dominant negative hnrnp mutant nuclear transport '\n",
      "                  'domain',\n",
      "                  'hnrnp mutant global inhibition viral mrna transcription '\n",
      "                  'genomic replication preferential inhibition replication '\n",
      "                  'defective interfering rna',\n",
      "                  'similar wild type hnrnp hnrnp mutant gene protein viral rna',\n",
      "                  'wild type hnrnp mutant protein bind kda cellular protein '\n",
      "                  'recruitment cellular protein hnrnp',\n",
      "                  'importance cellular factor viral rna dependent rna '\n",
      "                  'synthesis',\n",
      "                  'wild type mutant gene protein viral rna',\n",
      "                  'viral rna expression dominant negative mutant nuclear',\n",
      "                  'nuclear mrna',\n",
      "                  'mutant inhibition mrna transcription replication inhibition '\n",
      "                  'replication',\n",
      "                  'wild type mutant protein cellular protein rna synthesis',\n",
      "                  'cellular viral rna rna synthesis',\n",
      "                  'present study demonstrate transcription replication'],\n",
      "  'pr_title': ['heterogeneous nuclear ribonucleoprotein regulates rna '\n",
      "               'synthesis cytoplasmic virus'],\n",
      "  'publish_time': '2000-09-01',\n",
      "  'pubmed_id': '10970862',\n",
      "  's2_id': None,\n",
      "  'sha': '14e0cac6e86d62859e6c9f1351ab67466d89e5b3',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Heterogeneous nuclear ribonucleoprotein A1 regulates RNA synthesis '\n",
      "           'of a cytoplasmic virus',\n",
      "  'url': 'http://europepmc.org/articles/pmc302072?pdf=render',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'The UBA domain is a conserved sequence motif among '\n",
      "              'polyubiquitin binding proteins. For the first time, we '\n",
      "              'demonstrate a systematic, high throughput approach to '\n",
      "              'identification of UBA domain-interacting proteins from a '\n",
      "              'proteome-wide perspective. Using the rabbit reticulocyte lysate '\n",
      "              'in vitro expression cloning system, we have successfully '\n",
      "              'identified eleven proteins that interact with p62’s UBA domain, '\n",
      "              'and the majority of the eleven proteins are associated with '\n",
      "              'neurodegenerative disorders, such as Alzheimer’s disease. '\n",
      "              'Therefore, p62 may play a novel regulatory role through its UBA '\n",
      "              'domain. Our approach provides an easy route to the '\n",
      "              'characterization of UBA domain interacting proteins and its '\n",
      "              'application will unfold the important roles that the UBA domain '\n",
      "              'plays.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Pridgeon, Julia W.; Geetha, Thangiah; Wooten, Marie W.',\n",
      "  'cord_uid': 'tvxpckxo',\n",
      "  'doi': '10.1251/bpo66',\n",
      "  'journal': 'Biol Proced Online',\n",
      "  'license': 'no-cc',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/d09b79026117ec9faebba46a8d13aa9b23ec751e.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC302190.xml.json',\n",
      "  'pmcid': 'PMC302190',\n",
      "  'pr_abstract': ['uba domain',\n",
      "                  'first time uba domain protein',\n",
      "                  'eleven protein eleven protein',\n",
      "                  'uba domain',\n",
      "                  'approach uba domain',\n",
      "                  'approach uba domain protein uba domain',\n",
      "                  'protein uba domain protein',\n",
      "                  'first time approach uba domain protein',\n",
      "                  'uba domain protein',\n",
      "                  'uba domain'],\n",
      "  'pr_title': ['method identify uba domain interacting protein'],\n",
      "  'publish_time': '2003-12-12',\n",
      "  'pubmed_id': '14702098',\n",
      "  's2_id': None,\n",
      "  'sha': 'd09b79026117ec9faebba46a8d13aa9b23ec751e',\n",
      "  'source_x': 'PMC',\n",
      "  'title': \"A Method to Identify p62's UBA Domain Interacting Proteins\",\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC302190/',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'We examined the role of the microtubule cytoskeleton during '\n",
      "              'vaccinia virus infection. We found that newly assembled virus '\n",
      "              'particles accumulate in the vicinity of the '\n",
      "              'microtubule-organizing centre in a microtubule- and '\n",
      "              'dynein–dynactin complex-dependent fashion. Microtubules are '\n",
      "              'required for efficient intracellular mature virus (IMV) '\n",
      "              'formation and are essential for intracellular enveloped virus '\n",
      "              '(IEV) assembly. As infection proceeds, the microtubule '\n",
      "              'cytoskeleton becomes dramatically reorganized in a fashion '\n",
      "              'reminiscent of overexpression of microtubule-associated '\n",
      "              'proteins (MAPs). Consistent with this, we report that the '\n",
      "              'vaccinia proteins A10L and L4R have MAP-like properties and '\n",
      "              'mediate direct binding of viral cores to microtubules in vitro. '\n",
      "              'In addition, vaccinia infection also results in severe '\n",
      "              'reduction of proteins at the centrosome and loss of centrosomal '\n",
      "              'microtubule nucleation efficiency. This represents the first '\n",
      "              'example of viral-induced disruption of centrosome function. '\n",
      "              'Further studies with vaccinia will provide insights into the '\n",
      "              'role of microtubules during viral pathogenesis and regulation '\n",
      "              'of centrosome function.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Ploubidou, Aspasia; Moreau, Violaine; Ashman, Keith; Reckmann, '\n",
      "             'Inge; González, Cayetano; Way, Michael',\n",
      "  'cord_uid': 'mcuixluu',\n",
      "  'doi': '10.1093/emboj/19.15.3932',\n",
      "  'journal': 'The EMBO Journal',\n",
      "  'license': 'green-oa',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/44102e3e69e70ad2a73e753133283334ce1f8736.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC306617.xml.json',\n",
      "  'pmcid': 'PMC306617',\n",
      "  'pr_abstract': ['examined role microtubule cytoskeleton vaccinia virus '\n",
      "                  'infection',\n",
      "                  'newly assembled virus particle',\n",
      "                  'microtubule efficient intracellular mature virus imv '\n",
      "                  'formation essential intracellular virus assembly',\n",
      "                  'infection microtubule cytoskeleton fashion reminiscent '\n",
      "                  'overexpression microtubule associated protein',\n",
      "                  'consistent report vaccinia protein map property direct '\n",
      "                  'binding viral core microtubule',\n",
      "                  'addition vaccinia infection severe reduction protein loss '\n",
      "                  'efficiency',\n",
      "                  'first example function',\n",
      "                  'study vaccinia insight role microtubule viral pathogenesis '\n",
      "                  'regulation centrosome function',\n",
      "                  'role virus infection',\n",
      "                  'infection protein',\n",
      "                  'intracellular virus intracellular virus',\n",
      "                  'newly virus particle',\n",
      "                  'consistent protein map property direct viral',\n",
      "                  'represents first viral function',\n",
      "                  'infection reduction protein loss efficiency',\n",
      "                  'study insight role viral function'],\n",
      "  'pr_title': ['vaccinia virus infection disrupts microtubule organization '\n",
      "               'centrosome function'],\n",
      "  'publish_time': '2000-08-01',\n",
      "  'pubmed_id': '10921875',\n",
      "  's2_id': None,\n",
      "  'sha': '44102e3e69e70ad2a73e753133283334ce1f8736',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Vaccinia virus infection disrupts microtubule organization and '\n",
      "           'centrosome function',\n",
      "  'url': 'http://europepmc.org/articles/pmc306617?pdf=render',\n",
      "  'who_covidence_id': None},\n",
      " {'abstract': 'BACKGROUND: The present study aimed to provide information on '\n",
      "              'awareness of the attributable fraction of cancer causes among '\n",
      "              'the Japanese general population. METHODS: A nationwide '\n",
      "              'representative sample of 2,000 Japanese aged 20 or older was '\n",
      "              'asked about their perception and level of concern about various '\n",
      "              'environmental and genetic risk factors in relation to cancer '\n",
      "              'prevention, as a part of an Omnibus Survey. Interviews were '\n",
      "              'conducted with 1,355 subjects (609 men and 746 women). RESULTS: '\n",
      "              'Among 12 risk factor candidates, the attributable fraction of '\n",
      "              'cancer-causing viral and bacterial infection was considered '\n",
      "              'highest (51%), followed by that of tobacco smoking (43%), '\n",
      "              'stress (39%), and endocrine-disrupting chemicals (37%). On the '\n",
      "              'other hand, the attributable fractions of cancer by charred '\n",
      "              'fish and meat (21%) and alcohol drinking (22%) were considered '\n",
      "              'low compared with other risk factor candidates. For most risk '\n",
      "              'factors, attributable fraction responses were higher in women '\n",
      "              'than in men. As a whole, the subjects tended to respond with '\n",
      "              'higher values than those estimated by epidemiologic evidence in '\n",
      "              'the West. The attributable fraction of cancer speculated to be '\n",
      "              'genetically determined was 32%, while 36% of cancer was '\n",
      "              'considered preventable by improving lifestyle. CONCLUSION: Our '\n",
      "              'results suggest that awareness of the attributable fraction of '\n",
      "              'cancer causes in the Japanese general population tends to be '\n",
      "              'dominated by cancer-causing infection, occupational exposure, '\n",
      "              'air pollution and food additives rather than major lifestyle '\n",
      "              'factors such as diet.',\n",
      "  'arxiv_id': None,\n",
      "  'authors': 'Inoue, Manami; Iwasaki, Motoki; Otani, Tetsuya; Sasazuki, '\n",
      "             'Shizuka; Tsugane, Shoichiro',\n",
      "  'cord_uid': 'bnnl700a',\n",
      "  'doi': '10.1186/1471-2458-6-2',\n",
      "  'journal': 'BMC Public Health',\n",
      "  'license': 'cc-by',\n",
      "  'mag_id': None,\n",
      "  'pdf_json_files': 'document_parses/pdf_json/a78fd1b34372e1e54bf2a192d04aa36670cea307.json',\n",
      "  'pmc_json_files': 'document_parses/pmc_json/PMC1351169.xml.json',\n",
      "  'pmcid': 'PMC1351169',\n",
      "  'pr_abstract': ['background present study information awareness attributable '\n",
      "                  'fraction cancer japanese general population',\n",
      "                  'sample perception level concern risk factor prevention part',\n",
      "                  'subject men woman',\n",
      "                  'risk factor candidate attributable fraction cancer viral '\n",
      "                  'bacterial infection tobacco smoking stress endocrine',\n",
      "                  'hand fish meat alcohol drinking risk factor candidate',\n",
      "                  'risk factor response woman',\n",
      "                  'higher value estimated epidemiologic evidence',\n",
      "                  'attributable fraction cancer genetically determined cancer',\n",
      "                  'awareness attributable fraction cancer japanese general '\n",
      "                  'population infection exposure major lifestyle factor diet',\n",
      "                  'information cancer cause general population',\n",
      "                  'hand cancer considered risk factor candidate',\n",
      "                  'risk factor response higher woman',\n",
      "                  'risk factor candidate cancer infection considered',\n",
      "                  'woman',\n",
      "                  'higher',\n",
      "                  'cancer cause general population cancer infection exposure',\n",
      "                  'cancer cancer considered',\n",
      "                  'sample risk factor cancer prevention part'],\n",
      "  'pr_title': ['public awareness risk factor cancer among japanese general '\n",
      "               'population population based survey'],\n",
      "  'publish_time': '2006-01-10',\n",
      "  'pubmed_id': '16403223',\n",
      "  's2_id': None,\n",
      "  'sha': 'a78fd1b34372e1e54bf2a192d04aa36670cea307',\n",
      "  'source_x': 'PMC',\n",
      "  'title': 'Public awareness of risk factors for cancer among the Japanese '\n",
      "           'general population: A population-based survey',\n",
      "  'url': 'https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1351169/',\n",
      "  'who_covidence_id': None}]\n"
     ]
    }
   ],
   "source": [
    "documents=search(es, es_index=\"covid\",\n",
    "    query={\n",
    "          \"query\": {\n",
    "            \"match_all\":{}\n",
    "          }\n",
    "        })\n",
    "print(\"Retrieved documents:\")\n",
    "pprint.pprint(documents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
